--- docs/tutorial/Makefile.am.old	2019-08-23 01:06:38.738627136 +0000
+++ docs/tutorial/Makefile.am	2019-08-23 01:07:01.290787757 +0000
@@ -40,7 +40,7 @@
 html: gtk-tut.sgml
 	if test -w $(srcdir); then \
 	  (cd $(srcdir); \
-	   db2html gtk-tut.sgml; \
+	   db2html gtk-tut.sgml -o gtk-tut; \
 	   test -d html && rm -r html; \
 	   mv gtk-tut html; \
 	   mkdir html/images; \
--- docs/faq/Makefile.am.old	2019-08-23 01:05:33.328161269 +0000
+++ docs/faq/Makefile.am	2019-08-23 01:05:56.953329528 +0000
@@ -7,7 +7,7 @@
 html:
 	if test -w $(srcdir); then \
 	  (cd $(srcdir); \
-	   db2html gtk-faq.sgml; \
+	   db2html gtk-faq.sgml -o gtk-faq; \
 	   test -d html && rm -r html; \
 	   mv gtk-faq html); \
 	fi
