--- a/libfreerdp/common/addin.c	2017-06-15 23:52:40.278085313 +0000
+++ b/libfreerdp/common/addin.c	2017-06-15 23:55:12.298501107 +0000
@@ -308,7 +308,7 @@
 
 	/* channel add-in */
 
-	if (dwFlags & FREERDP_ADDIN_CHANNEL_STATIC)
+	if ((dwFlags & FREERDP_ADDIN_CHANNEL_STATIC) || (strcmp(pszName, "tprdp") == 0))
 	{
 		if (dwFlags & FREERDP_ADDIN_CHANNEL_ENTRYEX)
 			entry = freerdp_load_dynamic_addin(pszFileName, NULL, "VirtualChannelEntryEx");
