--- a/libfreerdp/common/addin.c
+++ b/libfreerdp/common/addin.c
@@ -204,7 +204,7 @@
 
 		/* channel add-in */
 
-	if (dwFlags & FREERDP_ADDIN_CHANNEL_STATIC)
+	if ((dwFlags & FREERDP_ADDIN_CHANNEL_STATIC) || (strcmp(pszName, "tprdp") == 0))
 		entry = freerdp_load_dynamic_addin(pszFileName, NULL, "VirtualChannelEntry");
 	else if (dwFlags & FREERDP_ADDIN_CHANNEL_DYNAMIC)
 		entry = freerdp_load_dynamic_addin(pszFileName, NULL, "DVCPluginEntry");
