--- plugins/digestmd5.c~	2008-11-08 18:28:21.000000000 +0000
+++ plugins/digestmd5.c	2008-11-08 18:28:50.000000000 +0000
@@ -2715,7 +2715,7 @@
 	"DIGEST-MD5",			/* mech_name */
 #ifdef WITH_RC4
 	128,				/* max_ssf */
-#elif WITH_DES
+#elif defined(WITH_DES)
 	112,
 #else 
 	1,
@@ -4034,7 +4034,7 @@
 	"DIGEST-MD5",
 #ifdef WITH_RC4				/* mech_name */
 	128,				/* max ssf */
-#elif WITH_DES
+#elif defined(WITH_DES)
 	112,
 #else
 	1,
