--- drivers/ata/pata_cs5536.c.orig	2014-05-26 08:53:09.459455512 +0000
+++ drivers/ata/pata_cs5536.c	2014-05-26 08:53:16.795714171 +0000
@@ -42,7 +42,7 @@
 
 #ifdef CONFIG_X86_32
 #include <asm/msr.h>
-static int use_msr;
+static int use_msr = 1;
 module_param_named(msr, use_msr, int, 0644);
 MODULE_PARM_DESC(msr, "Force using MSR to configure IDE function (Default: 0)");
 #else
@@ -195,6 +195,19 @@
 	cs5536_write(pdev, CAST, cast);
 }
 
+/*
+ * The mfgpt timer interrupt is running early, so we must keep the south bridge
+ * mmio always enabled. Otherwise we may race with the PCI configuration which
+ * may temporarily disable it. When that happens and the timer interrupt fires,
+ * we are not able to clear it and the system will hang.
+ */
+static void cs5536_isa_mmio_always_on(struct pci_dev *dev)
+{
+	dev->mmio_always_on = 1;
+}
+DECLARE_PCI_FIXUP_CLASS_EARLY(PCI_VENDOR_ID_AMD, PCI_DEVICE_ID_AMD_CS5536_ISA,
+	PCI_CLASS_BRIDGE_ISA, 8, cs5536_isa_mmio_always_on);
+
 /**
  *	cs5536_set_dmamode		-	DMA timing setup
  *	@ap: ATA interface
@@ -277,6 +290,10 @@
 
 	if (use_msr)
 		printk(KERN_ERR DRV_NAME ": Using MSR regs instead of PCI\n");
+		cs5536_write( dev,  CFG, 0x00010002 );
+		cs5536_write( dev,  DTC, 0x20200000 );
+		cs5536_write( dev, CAST, 0x20000000 );
+		cs5536_write( dev,  ETC, 0x00000000 );
 
 	cs5536_read(dev, CFG, &cfg);
 
