diff -urN linux-6.1.94.orig/drivers/video/fbdev/core/fbcon.c linux-6.1.94/drivers/video/fbdev/core/fbcon.c
--- linux-6.1.94.orig/drivers/video/fbdev/core/fbcon.c	2024-06-16 11:41:42.000000000 +0000
+++ linux-6.1.94/drivers/video/fbdev/core/fbcon.c	2024-06-20 17:55:13.051752880 +0000
@@ -569,6 +569,7 @@
 	if (fb_get_color_depth(&info->var, &info->fix) == 1)
 		erase &= ~0x400;
 	logo_height = fb_prepare_logo(info, ops->rotate);
+	logo_height += (info->var.yres/2)-(logo_height/2);
 	logo_lines = DIV_ROUND_UP(logo_height, vc->vc_font.height);
 	q = (unsigned short *) (vc->vc_origin +
 				vc->vc_size_row * rows);
@@ -1004,9 +1005,6 @@
 
 	info = fbcon_info_from_console(vc->vc_num);
 
-	if (logo_shown < 0 && console_loglevel <= CONSOLE_LOGLEVEL_QUIET)
-		logo_shown = FBCON_LOGO_DONTSHOW;
-
 	if (vc != svc || logo_shown == FBCON_LOGO_DONTSHOW ||
 	    (info->fix.type == FB_TYPE_TEXT))
 		logo = 0;
diff -urN linux-6.1.94.orig/drivers/video/fbdev/core/fbmem.c linux-6.1.94/drivers/video/fbdev/core/fbmem.c
--- linux-6.1.94.orig/drivers/video/fbdev/core/fbmem.c	2024-06-16 11:41:42.000000000 +0000
+++ linux-6.1.94/drivers/video/fbdev/core/fbmem.c	2024-06-20 17:47:20.285291098 +0000
@@ -501,7 +501,7 @@
 		fb_set_logo(info, logo, logo_new, fb_logo.depth);
 	}
 
-	if (fb_center_logo) {
+	if (true) {
 		int xres = info->var.xres;
 		int yres = info->var.yres;
 
