diff -Nru exim-4.77.orig/OS/Makefile-Base exim-4.77/OS/Makefile-Base
--- exim-4.77.orig/OS/Makefile-Base	2011-10-10 08:45:22.289284198 +0200
+++ exim-4.77/OS/Makefile-Base	2011-10-10 11:22:38.007450668 +0200
@@ -455,7 +455,7 @@
 
 # Update Exim's version information and build the version object.
 
-version.h::
+version.h:
 	@../scripts/reversion
 
 version.o: $(HDRS) cnumber.h version.h version.c
diff -Nru exim-4.77.orig/src/version.sh exim-4.77/src/version.sh
--- exim-4.77.orig/src/version.sh	2011-10-10 08:45:22.279284561 +0200
+++ exim-4.77/src/version.sh	2011-10-10 11:22:53.783546108 +0200
@@ -1,4 +1,4 @@
 # automatically generated file - see ../scripts/reversion
 EXIM_RELEASE_VERSION="4.77"
 EXIM_VARIANT_VERSION=""
-EXIM_COMPILE_NUMBER="1"
+EXIM_COMPILE_NUMBER="0"
