From 6caf15b6bd2a27f6cf955dfc0b44c245668a3e56 Mon Sep 17 00:00:00 2001
From: root <root@localhost.localdomain>
Date: Thu, 17 Apr 2014 04:32:49 +0000
Subject: [PATCH] 	modified:   src/via_id.c

---
 src/via_id.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/via_id.c b/src/via_id.c
index fe88114..fdfb66a 100644
--- a/src/via_id.c
+++ b/src/via_id.c
@@ -259,6 +259,7 @@ static struct ViaCardIdStruct ViaCardId[] = {
     {"Biostar Viotech 3200+",                 VIA_VX900,   0x1565, 0x120A, VIA_DEVICE_CRT},
     {"ASRock PV530",                          VIA_VX900,   0x1849, 0x7122, VIA_DEVICE_CRT},
     {"Fujitsu Futro A300",                    VIA_VX900,   0xA0A0, 0x080F, VIA_DEVICE_CRT},
+    {"Itona Md54",                            VIA_VX900,   0x1019, 0x7DAE, VIA_DEVICE_CRT},
 
     /* keep this */
     {NULL,                                    VIA_UNKNOWN, 0x0000, 0x0000, VIA_DEVICE_NONE}
-- 
1.9.1

