--- opt/Adobe/Reader9/bin/acroread.orig	2013-10-20 20:37:39.725415602 +0000
+++ opt/Adobe/Reader9/bin/acroread	2013-10-20 20:37:57.281869336 +0000
@@ -526,7 +526,7 @@
 if [ -z "$mozillaPath" ]; then
   MozPath="`gconftool-2 -g /desktop/gnome/url-handlers/http/command 2>/dev/null | cut -d ' ' -f1 2>/dev/null`"
   [ -x "${MozPath}" ] || MozPath="`Which "$MozPath"`"
-  [ -n "${MozPath}" ] || mozillaPath="`readlink "$MozPath" | xargs dirname`"
+  [ -n "${MozPath}" ] || mozillaPath="`readlink "$MozPath" | xargs dirname 2>/dev/null`"
   if [ -n "$mozillaPath" ] &&  $TestCmd $TestCmdOption "$mozillaPath/libgtkembedmoz.so" || $TestCmd $TestCmdOption "$mozillaPath/libxul.so" ; then
     MOZILLA_COMP_PATH="$mozillaPath"
     export MOZILLA_COMP_PATH
