--- src/notification.c.orig	2014-04-18 19:25:43.058047405 +0000
+++ src/notification.c	2014-04-18 19:28:03.776520475 +0000
@@ -184,10 +184,7 @@
             icon = freeme = g_filename_to_uri (icon_file, NULL, NULL);
           }
 
-          notif = notify_notification_new_with_status_icon (
-                              message[0] /* title */,
-                              message[1] /* summary */,
-                              icon, status_icon);
+          notif = NULL;
 
           g_strfreev (message);
           g_free (freeme);
