diff -ur a/hw/xfree86/common/xf86.h b/hw/xfree86/common/xf86.h
--- a/hw/xfree86/common/xf86.h	2014-09-21 08:55:24.000000000 +0000
+++ b/hw/xfree86/common/xf86.h	2014-11-12 17:21:34.337938083 +0000
@@ -215,7 +215,7 @@
 
 /* xf86Events.c */
 
-typedef struct _InputInfoRec *InputInfoPtr;
+struct _InputInfoRec;
 
 extern _X_EXPORT void SetTimeSinceLastInputEvent(void);
 extern _X_EXPORT void *xf86AddInputHandler(int fd, InputHandlerProc proc,
@@ -238,8 +238,8 @@
 extern _X_EXPORT Bool xf86VTOwner(void);
 extern _X_EXPORT void xf86VTLeave(void);
 extern _X_EXPORT void xf86VTEnter(void);
-extern _X_EXPORT void xf86EnableInputDeviceForVTSwitch(InputInfoPtr pInfo);
-extern _X_EXPORT void xf86DisableInputDeviceForVTSwitch(InputInfoPtr pInfo);
+extern _X_EXPORT void xf86EnableInputDeviceForVTSwitch(struct _InputInfoRec* pInfo);
+extern _X_EXPORT void xf86DisableInputDeviceForVTSwitch(struct _InputInfoRec* pInfo);
 
 /* xf86Helper.c */
 
diff -ur a/hw/xfree86/os-support/linux/lnx_init.c b/hw/xfree86/os-support/linux/lnx_init.c
--- a/hw/xfree86/os-support/linux/lnx_init.c	2014-09-21 08:56:23.000000000 +0000
+++ b/hw/xfree86/os-support/linux/lnx_init.c	2014-11-12 17:22:44.249937818 +0000
@@ -76,6 +76,7 @@
         FatalError("%s: VT_WAITACTIVE failed: %s\n", from, strerror(errno));
 }
 
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
 void
 xf86OpenConsole(void)
 {
@@ -166,7 +167,6 @@
 
         i = 0;
         while (vcs[i] != NULL) {
-#pragma GCC diagnostic ignored "-Wformat-nonliteral"
             snprintf(vtname, sizeof(vtname), vcs[i], xf86Info.vtno);    /* /dev/tty1-64 */
             if ((xf86Info.consoleFd = open(vtname, O_RDWR | O_NDELAY, 0)) >= 0)
                 break;
diff -ur a/os/log.c b/os/log.c
--- a/os/log.c	2014-09-21 08:56:23.000000000 +0000
+++ b/os/log.c	2014-11-12 17:03:14.609939415 +0000
@@ -188,6 +188,7 @@
  * %s, if present in the fname or backup strings, is expanded to the display
  * string.
  */
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
 
 const char *
 LogInit(const char *fname, const char *backup)
@@ -195,9 +196,6 @@
     char *logFileName = NULL;
 
     if (fname && *fname) {
-#if __GNUC__ > 4 || __GNUC__ == 4 && __GNUC_MINOR__ > 2
-#pragma GCC diagnostic ignored "-Wformat-nonliteral"
-#endif
         if (asprintf(&logFileName, fname, display) == -1)
             FatalError("Cannot allocate space for the log file name\n");
 
@@ -208,9 +206,6 @@
                 char *suffix;
                 char *oldLog;
 
-#if __GNUC__ > 4 || __GNUC__ == 4 && __GNUC_MINOR__ > 2
-#pragma GCC diagnostic ignored "-Wformat-nonliteral"
-#endif
                 if ((asprintf(&suffix, backup, display) == -1) ||
                     (asprintf(&oldLog, "%s%s", logFileName, suffix) == -1))
                     FatalError("Cannot allocate space for the log file name\n");
diff -ur a/test/signal-logging.c b/test/signal-logging.c
--- a/test/signal-logging.c	2014-09-21 08:56:23.000000000 +0000
+++ b/test/signal-logging.c	2014-11-12 17:24:38.705937821 +0000
@@ -113,12 +113,13 @@
                          -1, -2.00, -0.6023, -1203.30
                         };
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Woverflow"
+#pragma GCC diagnostic pop
 static void
 number_formatting(void)
 {
     int i;
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Woverflow"
     long unsigned int unsigned_tests[] = { 0,/* Zero */
                                            5, /* Single digit number */
                                            12, /* Two digit decimal number */
@@ -141,7 +142,6 @@
                                 -0x15D027BF211B37A, /* Large > 32 bit number */
                                 -0x7FFFFFFFFFFFFFFF, /* Maximum 64-bit signed number */
     } ;
-#pragma GCC diagnostic pop
 
     for (i = 0; i < sizeof(unsigned_tests) / sizeof(unsigned_tests[0]); i++)
         assert(check_number_format_test(unsigned_tests[i]));
