diff -Nru parted-3.1.orig/parted/ui.c parted-3.1/parted/ui.c
--- parted-3.1.orig/parted/ui.c	2014-03-04 11:20:01.556849738 +0100
+++ parted-3.1/parted/ui.c	2014-03-04 11:21:36.810139757 +0100
@@ -1474,7 +1474,7 @@
 #ifdef HAVE_LIBREADLINE
   if (!opt_script_mode) {
     rl_initialize ();
-    rl_attempted_completion_function = (CPPFunction*) complete_function;
+    rl_attempted_completion_function = (rl_completion_func_t *) complete_function;
     readline_state.in_readline = 0;
   }
 #endif
