--- src/xminicom.old	2013-02-22 05:16:28.448927119 +0000
+++ src/xminicom	2013-02-22 05:18:35.923931121 +0000
@@ -59,5 +59,9 @@
   exit 1
 fi
 
-echo "xminicom: rxvt, color_xterm or xterm NOT found!" 1>&2
+if findcmd xfce4-terminal; then
+	exec $result -T minicom -I minicom --geometry 80x25 -x minicom -m -c on $*
+fi
+
+echo "xminicom: rxvt, color_xterm , xfce4-terminal or xterm NOT found!" 1>&2
 exit 1
