diff -ru ppp-2.4.3/pppd/Makefile.linux ppp-2.4.3-new/pppd/Makefile.linux
--- ppp-2.4.3/pppd/Makefile.linux	2004-11-13 13:02:22.000000000 +0100
+++ ppp-2.4.3-new/pppd/Makefile.linux	2004-12-12 00:05:13.582252968 +0100
@@ -48,7 +48,7 @@
 # Uncomment the next line to include support for PPP packet filtering.
 # This requires that the libpcap library and headers be installed
 # and that the kernel driver support PPP packet filtering.
-FILTER=y
+#FILTER=y
 
 # Uncomment the next line to enable multilink PPP (enabled by default)
 # Linux distributions: Please leave multilink ENABLED in your builds
