--- 10_linux.in.orig	2012-09-05 13:34:26.947842426 -0500
+++ 10_linux.in	2012-09-05 13:34:55.967842887 -0500
@@ -193,6 +193,7 @@
   initrd=
   for i in "initrd.img-${version}" "initrd-${version}.img" "initrd-${version}.gz" \
 	   "initrd-${version}" "initramfs-${version}.img" \
+	   "initramfs-${version}.gz" \
 	   "initrd.img-${alt_version}" "initrd-${alt_version}.img" \
 	   "initrd-${alt_version}" "initramfs-${alt_version}.img" \
 	   "initramfs-genkernel-${version}" \
