--- opensc.conf.orig	2011-10-24 12:49:41.944009862 +0000
+++ opensc.conf	2011-10-24 13:41:11.183389056 +0000
@@ -12,14 +12,14 @@
 	# A greater value means more debug info.
 	# Default: 0
 	#
-	debug = 0;
+	debug = 2;
 
 	# The file to which debug output will be written
 	#
 	# Special values 'stdout' and 'stderr' are recognized.
 	# Default: stderr
 	#
-	# debug_file = /tmp/opensc-debug.log;
+	debug_file = /tmp/opensc-debug.log;
 	# debug_file = "C:\Documents and Settings\All Users\Documents\opensc-debug.log";
 
 	# PKCS#15 initialization / personalization
@@ -74,7 +74,7 @@
 		#
 		# Use specific pcsc provider.
 		# Default: libpcsclite.so.1
-		# provider_library = libpcsclite.so.1
+		provider_library = /lib/libpcsclite.so.1
 	}
 
 	# Options for OpenCT support
