diff -Nru qt-x11-free-3.3.4-orig/tools/qvfb/qvfbview.cpp qt-x11-free-3.3.4/tools/qvfb/qvfbview.cpp
--- qt-x11-free-3.3.4-orig/tools/qvfb/qvfbview.cpp	2005-10-31 21:49:58.000000000 +0000
+++ qt-x11-free-3.3.4/tools/qvfb/qvfbview.cpp	2005-10-31 21:51:52.000000000 +0000
@@ -115,7 +115,7 @@
 	data = (unsigned char *)shmat( shmId, 0, 0 );
     }
 
-    if ( (int)data == -1 )
+    if ( (long)data == -1 )
 	qFatal( "Cannot attach to shared memory" );
 
     hdr = (QVFbHeader *)data;
