diff -pruN boost-jam-3.1.17.orig/Jambase boost-jam-3.1.17/Jambase
--- boost-jam-3.1.17.orig/Jambase	2008-11-30 01:32:38.692291050 +1100
+++ boost-jam-3.1.17/Jambase	2008-11-30 01:36:31.488291745 +1100
@@ -50,7 +50,7 @@ BOOST_BUILD_PATH.user-value = $(BOOST_BU
 # Policy.
 if ! $(BOOST_BUILD_PATH) && $(UNIX)
 {
-    BOOST_BUILD_PATH = /usr/share/boost-build ;
+    BOOST_BUILD_PATH = /usr/share/boost/v2 ;
 }
 
 
diff -pruN boost-jam-3.1.17.orig/jambase.c boost-jam-3.1.17/jambase.c
--- boost-jam-3.1.17.orig/jambase.c	2008-11-30 01:32:38.694291102 +1100
+++ boost-jam-3.1.17/jambase.c	2008-11-30 01:36:31.490291290 +1100
@@ -21,7 +21,7 @@ char *jambase[] = {
 "BOOST_BUILD_PATH.user-value = $(BOOST_BUILD_PATH) ;\n",
 "if ! $(BOOST_BUILD_PATH) && $(UNIX)\n",
 "{\n",
-"BOOST_BUILD_PATH = /usr/share/boost-build ;\n",
+"BOOST_BUILD_PATH = /usr/share/boost/v2 ;\n",
 "}\n",
 "rule _poke ( module-name ? : variables + : value * )\n",
 "{\n",
