Fix a stupid bashism
--- configure.orig	2006-05-02 02:01:12.637169936 +0200
+++ configure	2006-05-02 02:01:30.861399432 +0200
@@ -5217,7 +5217,7 @@ if test "${with_app_defaults+set}" = set
   withval="$with_app_defaults"
 
 else
-  withval="${appsdir-'\$(exec_prefix)/lib/X11/app-defaults'}"
+  withval="${appsdir-'\$\(exec_prefix\)/lib/X11/app-defaults'}"
 fi;
 case ".$withval" in #(vi
 .\$\(*\)*|.\'*\'*) #(vi
