--- mk/syslinux.mk.orig	2014-11-21 17:49:40.778816445 +0000
+++ mk/syslinux.mk	2014-11-21 17:51:10.199067070 +0000
@@ -46,20 +46,20 @@
 
 CHMOD	 = chmod
 
-CC	 = gcc
+CC	 = $(CROSS_COMPILE)gcc
 gcc_ok   = $(shell tmpf=gcc_ok.$$$$.tmp; \
 		   if $(CC) $(GCCOPT) $(1) -c $(topdir)/dummy.c \
 			-o $$tmpf 2>/dev/null ; \
 		   then echo '$(1)'; else echo '$(2)'; fi; \
 		   rm -f $$tmpf)
 
-LD	 = ld
-OBJDUMP	 = objdump
-OBJCOPY  = objcopy
-STRIP    = strip
-AR       = ar
-NM       = nm
-RANLIB   = ranlib
+LD	 = $(CROSS_COMPILE)ld
+OBJDUMP	 = $(CROSS_COMPILE)objdump
+OBJCOPY  = $(CROSS_COMPILE)objcopy
+STRIP    = $(CROSS_COMPILE)strip
+AR       = $(CROSS_COMPILE)ar
+NM       = $(CROSS_COMPILE)nm
+RANLIB   = $(CROSS_COMPILE)ranlib
 STRIP	 = strip
 GZIPPROG = gzip
 XZ	 = xz
