diff -up libproxy-0.4.10/libproxy/cmake/modules/pacrunner_mozjs.cmk.orig 
libproxy-0.4.10/libproxy/cmake/modules/pacrunner_mozjs.cmk
--- libproxy-0.4.10/libproxy/cmake/modules/pacrunner_mozjs.cmk.orig	2012-10-02 16:20:40.000000000 +0200
+++ libproxy-0.4.10/libproxy/cmake/modules/pacrunner_mozjs.cmk	2012-10-17 00:36:43.645579472 +0200
@@ -9,7 +9,7 @@ if(WIN32)
 elseif(NOT APPLE)
   option(WITH_MOZJS "Search for MOZJS package" ON)
   if (WITH_MOZJS)
-    pkg_search_module(MOZJS mozjs185>=1.8.5)
+    pkg_search_module(MOZJS mozjs185)
     if(MOZJS_FOUND)
       include_directories(${MOZJS_INCLUDE_DIRS})
       link_directories(${MOZJS_LIBRARY_DIRS})
