--- a/conf/cups-files.conf.in	2016-05-24 16:44:18.639114820 +0000
+++ b/conf/cups-files.conf.in	2016-05-24 16:44:59.863144890 +0000
@@ -25,6 +25,10 @@
 # Do we allow file: device URIs other than to /dev/null?
 #FileDevice No
 
+# Location of ssl certificates
+ServerCertificate /etc/ssl/certs/cups.crt
+ServerKey /etc/ssl/keys/cups.key
+
 # Permissions for configuration and log files...
 #ConfigFilePerm 0@CUPS_CONFIG_FILE_PERM@
 #LogFilePerm 0@CUPS_LOG_FILE_PERM@
