diff -Nru cups-1.6.2.orig/conf/cups-files.conf.in cups-1.6.2/conf/cups-files.conf.in
--- cups-1.6.2.orig/conf/cups-files.conf.in	2013-03-18 18:52:05.834379900 +0100
+++ cups-1.6.2/conf/cups-files.conf.in	2013-03-18 19:13:54.540806821 +0100
@@ -74,10 +74,10 @@
 #ServerBin @CUPS_SERVERBIN@
 
 # SSL/TLS certificate for the scheduler...
-#ServerCertificate @CUPS_SERVERCERT@
+ServerCertificate /etc/ssl/certs/cups.crt
 
 # SSL/TLS private key for the scheduler...
-#ServerKey @CUPS_SERVERKEY@
+ServerKey /etc/ssl/keys/cups.key
 
 # Location of other configuration files...
 #ServerRoot @CUPS_SERVERROOT@
