diff -urNp libunique-1.1.6.old/unique/Makefile.am libunique-1.1.6/unique/Makefile.am
--- libunique-1.1.6.old/unique/Makefile.am	2011-07-06 10:54:38.403127656 +0100
+++ libunique-1.1.6/unique/Makefile.am	2011-07-06 10:55:00.354127659 +0100
@@ -19,11 +19,8 @@ INCLUDES = -I$(top_srcdir)
 AM_CPPFLAGS = \
 	-DG_DISABLE_SINGLE_INCLUDES	\
 	-DGTK_DISABLE_SINGLE_INCLUDES	\
-	-DG_DISABLE_DEPRECATED		\
-	-DGTK_DISABLE_DEPRECATED	\
 	-DG_LOG_DOMAIN=\"Unique\"	\
 	-DPREFIX=\""$(prefix)"\"	\
-	$(DISABLE_DEPRECATED_FLAGS)	\
 	$(UNIQUE_DEBUG_CFLAGS)		\
 	$(NULL)
 
diff -urNp libunique-1.1.6.old/unique/Makefile.in libunique-1.1.6/unique/Makefile.in
--- libunique-1.1.6.old/unique/Makefile.in	2011-07-06 11:01:36.306127549 +0100
+++ libunique-1.1.6/unique/Makefile.in	2011-07-06 11:01:58.311127546 +0100
@@ -394,11 +394,8 @@ INCLUDES = -I$(top_srcdir)
 AM_CPPFLAGS = \
 	-DG_DISABLE_SINGLE_INCLUDES	\
 	-DGTK_DISABLE_SINGLE_INCLUDES	\
-	-DG_DISABLE_DEPRECATED		\
-	-DGTK_DISABLE_DEPRECATED	\
 	-DG_LOG_DOMAIN=\"Unique\"	\
 	-DPREFIX=\""$(prefix)"\"	\
-	$(DISABLE_DEPRECATED_FLAGS)	\
 	$(UNIQUE_DEBUG_CFLAGS)		\
 	$(NULL)
 
