--- services/plugins/resolutionSet/resolutionX11.c.orig	2014-04-18 23:44:01.283154934 +0000
+++ services/plugins/resolutionSet/resolutionX11.c	2014-04-18 23:44:54.656161320 +0000
@@ -605,6 +605,7 @@
    char *argv[] = {"", NULL};
    GtkWidget *wnd;
    Display *display;
+   gtk_disable_setlocale();
    gtk_init(&argc, (char ***) &argv);
    wnd = gtk_invisible_new();
    display = GDK_WINDOW_XDISPLAY(wnd->window);
--- services/plugins/desktopEvents/x11Lock.c.orig	2014-04-18 23:46:12.828100476 +0000
+++ services/plugins/desktopEvents/x11Lock.c	2014-04-18 23:46:48.073424822 +0000
@@ -365,6 +365,7 @@
    argv[0] = VMUSER_TITLE;
 
    /* XXX: is calling gtk_init() multiple times safe? */
+   gtk_disable_setlocale();
    gtk_init(&argc, (char ***) &argv);
 
    if (!AcquireDisplayLock()) {
