--- a/lib/include/vmware/tools/log.h	2014-06-04 13:01:54.879000679 +0000
+++ b/lib/include/vmware/tools/log.h	2014-06-04 13:02:11.995662720 +0000
@@ -121,6 +121,7 @@
 #  define FUNC __FUNCTION__
 #endif
 
+#ifndef g_info
 /*
  *******************************************************************************
  * g_info --                                                              */ /**
@@ -135,6 +136,7 @@
  */
 
 #define g_info(fmt, ...) g_log(G_LOG_DOMAIN, G_LOG_LEVEL_INFO, fmt, ## __VA_ARGS__)
+#endif
 
 
 /*
