--- open-vm-tools.sept-update/modules/linux/vmblock/linux/inode.c	2014-03-31 18:49:14.843009422 +0100
+++ open-vm-tools.sept-update/modules/linux/vmblock/linux/inode.c	2014-03-31 18:50:52.617009677 +0100
@@ -135,7 +135,12 @@ InodeOpLookup(struct inode *dir,      //
    inode->i_size = INODE_TO_IINFO(inode)->nameLen;
    inode->i_version = 1;
    inode->i_atime = inode->i_mtime = inode->i_ctime = CURRENT_TIME;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 14, 0)
    inode->i_uid = inode->i_gid = 0;
+#else
+   inode->i_uid = GLOBAL_ROOT_UID;
+   inode->i_gid = GLOBAL_ROOT_GID;
+#endif
    inode->i_op = &LinkInodeOps;
 
    d_add(dentry, inode);
--- open-vm-tools.sept-update/lib/include/vm_assert.h	2014-03-31 18:49:14.845009422 +0100
+++ open-vm-tools.sept-update/lib/include/vm_assert.h	2014-03-31 18:49:21.390009439 +0100
@@ -256,6 +256,7 @@ void WarningThrottled(uint32 *count, con
                                        __FILE__, __LINE__, __FUNCTION__, \
                                        _fix))
 #else
+   #undef DEPRECATED /* in <linux/printk.h> since 3.14.0 */
    #define DEPRECATED(_fix) do {} while (0)
 #endif
 
