diff --git a/src/plugins/canbus/socketcan/socketcanbackend.cpp b/src/plugins/canbus/socketcan/socketcanbackend.cpp
index 74b0d1d..a2da146 100644
--- a/qtserialbus/src/plugins/canbus/socketcan/socketcanbackend.cpp
+++ b/qtserialbus/src/plugins/canbus/socketcan/socketcanbackend.cpp
@@ -45,6 +45,7 @@
 
 #include <linux/can/error.h>
 #include <linux/can/raw.h>
+#include <linux/sockios.h>
 #include <errno.h>
 #include <unistd.h>
 #include <net/if.h>

