--- syntax/sh.nanorc.orig	2019-02-08 18:23:06.461112007 +0000
+++ syntax/sh.nanorc	2019-02-08 18:23:24.262266815 +0000
@@ -1,6 +1,6 @@
 ## Here is an example for Bourne shell scripts.
 
-syntax sh "(\.sh|\.(a|ba|c|da|k|mk|pdk|tc|z)(sh|shrc|sh_profile)|/(etc/|\.)profile)$"
+syntax sh "(\.sh|\.(a|ba|c|da|k|mk|pdk|tc|z)(sh|shrc|sh_profile)|/(etc/|\.)profile)$" "bash" "\.bash$" "conf" "\.conf$" "Pkgfile" "\.functions" "finalize"
 header "^#!.*/(((a|ba|c|da|k|mk|pdk|tc|z)?sh)|(busybox|env) +sh|openrc-run|runscript)"
 magic "(POSIX|Bourne-Again) shell script.*text"
 linter dash -n
