--- doc/syntax/sh.nanorc.orig	2016-12-01 04:30:46.513699642 +0000
+++ doc/syntax/sh.nanorc	2016-12-01 04:30:57.945980637 +0000
@@ -1,6 +1,6 @@
 ## Here is an example for Bourne shell scripts.
 
-syntax "sh" "\.sh$"
+syntax "sh" "\.sh$" "bash" "\.bash$" "conf" "\.conf$" "Pkgfile" "\.functions"
 header "^#!.*((ba|da|k|pdk)?sh[-0-9_]*|openrc-run|runscript)"
 magic "(POSIX|Bourne.*) shell script text"
 linter dash -n
