--- syntax/sh.nanorc.orig	2022-08-23 16:52:55.837775770 +0000
+++ syntax/sh.nanorc	2022-08-23 16:53:17.544706355 +0000
@@ -1,6 +1,6 @@
 ## Syntax highlighting for Bourne shell scripts.
 
-syntax sh "(\.sh|(\.|/)(a|ba|c|da|k|mk|pdk|tc|z)sh(rc|_profile)?|/(etc/|\.)profile)$"
+syntax sh "(\.sh|(\.|/)(a|ba|c|da|k|mk|pdk|tc|z)sh(rc|_profile)?|/(etc/|\.)profile)$" "bash" "\.bash$" "conf" "\.conf$" "Pkgfile" "\.functions" "finalize"
 header "^#!.*/((env[[:blank:]]+)?((a|ba|c|da|k|mk|pdk|tc|z)?sh)|busybox[[:blank:]]+sh|openrc-run|runscript)\>"
 header "-\*-.*shell-script.*-\*-"
 magic "(POSIX|Bourne-Again) shell script.*text"
