--- a/src/devicewidget.h	2014-06-18 16:03:35.597855977 +0000
+++ b/src/devicewidget.h	2014-06-18 16:09:30.492855250 +0000
@@ -98,7 +98,6 @@
     Gtk::HBox *portSelect, *offsetSelect;
     Gtk::ComboBox *portList;
     Glib::RefPtr<Gtk::ListStore> treeModel;
-    Glib::RefPtr<Gtk::Adjustment> offsetAdjustment;
 
 private:
     Glib::ustring mDeviceType;
--- a/src/devicewidget.cc	2014-06-18 16:02:17.125855839 +0000
+++ b/src/devicewidget.cc	2014-06-18 16:09:21.086855148 +0000
@@ -63,7 +63,7 @@
     for (unsigned i = 0; i < PA_CHANNELS_MAX; i++)
         channelWidgets[i] = NULL;
 
-    offsetAdjustment = Gtk::Adjustment::create(0.0, -2000.0, 2000.0, 10.0, 50.0, 0.0);
+    Gtk::Adjustment offsetAdjustment(0.0, -2000.0, 2000.0, 10.0, 50.0, 0.0);
     offsetButton->configure(offsetAdjustment, 0, 2);
 }
 
