diff -Nur LVM2.2.02.106.orig/scripts/dm_event_systemd_red_hat.socket.in LVM2.2.02.106/scripts/dm_event_systemd_red_hat.socket.in
--- LVM2.2.02.106.orig/scripts/dm_event_systemd_red_hat.socket.in	2014-04-10 17:38:46.000000000 +0200
+++ LVM2.2.02.106/scripts/dm_event_systemd_red_hat.socket.in	2014-04-26 14:21:10.097269315 +0200
@@ -8,6 +8,3 @@
 ListenFIFO=@DEFAULT_DM_RUN_DIR@/dmeventd-client
 SocketMode=0600
 RemoveOnStop=true
-
-[Install]
-WantedBy=sockets.target
diff -Nur LVM2.2.02.106.orig/scripts/lvm2_lvmetad_systemd_red_hat.socket.in LVM2.2.02.106/scripts/lvm2_lvmetad_systemd_red_hat.socket.in
--- LVM2.2.02.106.orig/scripts/lvm2_lvmetad_systemd_red_hat.socket.in	2014-04-10 17:38:46.000000000 +0200
+++ LVM2.2.02.106/scripts/lvm2_lvmetad_systemd_red_hat.socket.in	2014-04-26 14:21:15.287156152 +0200
@@ -7,6 +7,3 @@
 ListenStream=@DEFAULT_RUN_DIR@/lvmetad.socket
 SocketMode=0600
 RemoveOnStop=true
-
-[Install]
-WantedBy=sysinit.target
