commit 07d89829cce4527c7614a34642d4b2c2ef5d6005
Author: Will Estes <westes575@gmail.com>
Date:   Fri Dec 11 09:45:14 2015 -0500

    Commented in C style in skeleton; fixed sf#195

diff --git a/src/flex.skl b/src/flex.skl
index b63c8c8..40287b6 100644
--- a/src/flex.skl
+++ b/src/flex.skl
@@ -2350,7 +2350,7 @@ void yyFlexLexer::yyensure_buffer_stack(void)
 		 * scanner will even need a stack. We use 2 instead of 1 to avoid an
 		 * immediate realloc on the next call.
          */
-		num_to_alloc = 1; // After all that talk, this was set to 1 anyways...
+      num_to_alloc = 1; /* After all that talk, this was set to 1 anyways... */
 		YY_G(yy_buffer_stack) = (struct yy_buffer_state**)yyalloc
 								(num_to_alloc * sizeof(struct yy_buffer_state*)
 								M4_YY_CALL_LAST_ARG);
