--- traydevice-1.5.2/setup.py.orig	2010-06-19 12:54:32.000000000 +0000
+++ traydevice-1.5.2/setup.py	2012-05-10 19:17:55.544602465 +0000
@@ -231,6 +231,7 @@
         shutil.copy(source, man_tmp_dir)
         patched_file = join(man_tmp_dir, basename(source))
         patch_file(self.patch_line, patched_file)
+        return
         exe = subprocess.Popen(
             ["docbook2man", abspath(patched_file)], cwd=man_dir)
         exe.communicate()
