diff -u -r meanwhile-1.0.2.orig/src/channel.c meanwhile-1.0.2/src/channel.c
--- meanwhile-1.0.2.orig/src/channel.c	2012-06-25 22:26:37.012942968 +0900
+++ meanwhile-1.0.2/src/channel.c	2012-06-25 22:32:20.830934685 +0900
@@ -19,8 +19,6 @@
 */
 
 #include <glib.h>
-#include <glib/ghash.h>
-#include <glib/glist.h>
 #include <string.h>
 
 #include "mw_channel.h"
diff -u -r meanwhile-1.0.2.orig/src/mw_debug.c meanwhile-1.0.2/src/mw_debug.c
--- meanwhile-1.0.2.orig/src/mw_debug.c	2012-06-25 22:26:37.011942968 +0900
+++ meanwhile-1.0.2/src/mw_debug.c	2012-06-25 22:32:20.845934685 +0900
@@ -19,7 +19,6 @@
 */
 
 
-#include <glib/gstring.h>
 
 #include "mw_debug.h"
 
diff -u -r meanwhile-1.0.2.orig/src/mw_message.h meanwhile-1.0.2/src/mw_message.h
--- meanwhile-1.0.2.orig/src/mw_message.h	2012-06-25 22:26:37.012942968 +0900
+++ meanwhile-1.0.2/src/mw_message.h	2012-06-25 22:32:20.847934685 +0900
@@ -22,7 +22,6 @@
 #define _MW_MESSAGE_H
 
 
-#include <glib/glist.h>
 #include "mw_common.h"
 
 
diff -u -r meanwhile-1.0.2.orig/src/mw_srvc_conf.h meanwhile-1.0.2/src/mw_srvc_conf.h
--- meanwhile-1.0.2.orig/src/mw_srvc_conf.h	2012-06-25 22:26:37.012942968 +0900
+++ meanwhile-1.0.2/src/mw_srvc_conf.h	2012-06-25 22:32:20.852934683 +0900
@@ -22,7 +22,6 @@
 #define _MW_SRVC_CONF_H
 
 
-#include <glib/glist.h>
 #include "mw_common.h"
 
 
diff -u -r meanwhile-1.0.2.orig/src/mw_srvc_dir.h meanwhile-1.0.2/src/mw_srvc_dir.h
--- meanwhile-1.0.2.orig/src/mw_srvc_dir.h	2012-06-25 22:26:37.011942968 +0900
+++ meanwhile-1.0.2/src/mw_srvc_dir.h	2012-06-25 22:32:20.852934683 +0900
@@ -22,7 +22,6 @@
 
 
 #include <glib.h>
-#include <glib/glist.h>
 
 
 #ifdef __cplusplus
diff -u -r meanwhile-1.0.2.orig/src/mw_srvc_place.h meanwhile-1.0.2/src/mw_srvc_place.h
--- meanwhile-1.0.2.orig/src/mw_srvc_place.h	2012-06-25 22:26:37.012942968 +0900
+++ meanwhile-1.0.2/src/mw_srvc_place.h	2012-06-25 22:32:20.855934685 +0900
@@ -22,7 +22,6 @@
 #define _MW_SRVC_PLACE_H
 
 
-#include <glib/glist.h>
 #include "mw_common.h"
 
 
diff -u -r meanwhile-1.0.2.orig/src/mw_srvc_resolve.h meanwhile-1.0.2/src/mw_srvc_resolve.h
--- meanwhile-1.0.2.orig/src/mw_srvc_resolve.h	2012-06-25 22:26:37.011942968 +0900
+++ meanwhile-1.0.2/src/mw_srvc_resolve.h	2012-06-25 22:32:20.855934685 +0900
@@ -23,7 +23,6 @@
 
 
 #include <glib.h>
-#include <glib/glist.h>
 
 
 #ifdef __cplusplus
diff -u -r meanwhile-1.0.2.orig/src/mw_st_list.h meanwhile-1.0.2/src/mw_st_list.h
--- meanwhile-1.0.2.orig/src/mw_st_list.h	2012-06-25 22:26:37.012942968 +0900
+++ meanwhile-1.0.2/src/mw_st_list.h	2012-06-25 22:32:20.857934685 +0900
@@ -30,7 +30,6 @@
 
 
 #include <glib.h>
-#include <glib/glist.h>
 #include "mw_common.h"
 
 
diff -u -r meanwhile-1.0.2.orig/src/mw_util.h meanwhile-1.0.2/src/mw_util.h
--- meanwhile-1.0.2.orig/src/mw_util.h	2012-06-25 22:26:37.012942968 +0900
+++ meanwhile-1.0.2/src/mw_util.h	2012-06-25 22:32:20.857934685 +0900
@@ -23,8 +23,6 @@
 
 
 #include <glib.h>
-#include <glib/ghash.h>
-#include <glib/glist.h>
 
 
 #define map_guint_new() \
diff -u -r meanwhile-1.0.2.orig/src/srvc_aware.c meanwhile-1.0.2/src/srvc_aware.c
--- meanwhile-1.0.2.orig/src/srvc_aware.c	2012-06-25 22:26:37.012942968 +0900
+++ meanwhile-1.0.2/src/srvc_aware.c	2012-06-25 22:32:20.866934685 +0900
@@ -19,8 +19,6 @@
 */
 
 #include <glib.h>
-#include <glib/ghash.h>
-#include <glib/glist.h>
 #include <string.h>
 
 #include "mw_channel.h"
diff -u -r meanwhile-1.0.2.orig/src/srvc_conf.c meanwhile-1.0.2/src/srvc_conf.c
--- meanwhile-1.0.2.orig/src/srvc_conf.c	2012-06-25 22:26:37.012942968 +0900
+++ meanwhile-1.0.2/src/srvc_conf.c	2012-06-25 22:32:20.869934685 +0900
@@ -19,8 +19,6 @@
 */
 
 #include <glib.h>
-#include <glib/ghash.h>
-#include <glib/glist.h>
 
 #include <stdio.h>
 #include <stdlib.h>
diff -u -r meanwhile-1.0.2.orig/src/srvc_dir.c meanwhile-1.0.2/src/srvc_dir.c
--- meanwhile-1.0.2.orig/src/srvc_dir.c	2012-06-25 22:26:37.011942968 +0900
+++ meanwhile-1.0.2/src/srvc_dir.c	2012-06-25 22:32:20.871934685 +0900
@@ -18,7 +18,6 @@
   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
-#include <glib/ghash.h>
 
 #include "mw_channel.h"
 #include "mw_common.h"
diff -u -r meanwhile-1.0.2.orig/src/srvc_ft.c meanwhile-1.0.2/src/srvc_ft.c
--- meanwhile-1.0.2.orig/src/srvc_ft.c	2012-06-25 22:26:37.012942968 +0900
+++ meanwhile-1.0.2/src/srvc_ft.c	2012-06-25 22:32:20.873934685 +0900
@@ -19,7 +19,6 @@
 */
 
 
-#include <glib/glist.h>
 
 #include "mw_channel.h"
 #include "mw_common.h"
diff -u -r meanwhile-1.0.2.orig/src/srvc_im.c meanwhile-1.0.2/src/srvc_im.c
--- meanwhile-1.0.2.orig/src/srvc_im.c	2012-06-25 22:26:37.012942968 +0900
+++ meanwhile-1.0.2/src/srvc_im.c	2012-06-25 22:32:20.877934685 +0900
@@ -19,7 +19,6 @@
 */
 
 #include <glib.h>
-#include <glib/glist.h>
 #include <string.h>
 
 #include "mw_channel.h"
diff -u -r meanwhile-1.0.2.orig/src/srvc_place.c meanwhile-1.0.2/src/srvc_place.c
--- meanwhile-1.0.2.orig/src/srvc_place.c	2012-06-25 22:26:37.011942968 +0900
+++ meanwhile-1.0.2/src/srvc_place.c	2012-06-25 22:32:20.880934685 +0900
@@ -19,8 +19,6 @@
 */
 
 #include <glib.h>
-#include <glib/ghash.h>
-#include <glib/glist.h>
 
 #include <stdio.h>
 #include <stdlib.h>
diff -u -r meanwhile-1.0.2.orig/src/srvc_resolve.c meanwhile-1.0.2/src/srvc_resolve.c
--- meanwhile-1.0.2.orig/src/srvc_resolve.c	2012-06-25 22:26:37.012942968 +0900
+++ meanwhile-1.0.2/src/srvc_resolve.c	2012-06-25 22:32:20.881934685 +0900
@@ -18,7 +18,6 @@
   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
-#include <glib/ghash.h>
 
 #include "mw_channel.h"
 #include "mw_common.h"
diff -u -r meanwhile-1.0.2.orig/src/srvc_store.c meanwhile-1.0.2/src/srvc_store.c
--- meanwhile-1.0.2.orig/src/srvc_store.c	2012-06-25 22:26:37.011942968 +0900
+++ meanwhile-1.0.2/src/srvc_store.c	2012-06-25 22:32:20.883934685 +0900
@@ -18,7 +18,6 @@
   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
-#include <glib/glist.h>
 
 #include "mw_channel.h"
 #include "mw_debug.h"
diff -u -r meanwhile-1.0.2.orig/src/st_list.c meanwhile-1.0.2/src/st_list.c
--- meanwhile-1.0.2.orig/src/st_list.c	2012-06-25 22:26:37.012942968 +0900
+++ meanwhile-1.0.2/src/st_list.c	2012-06-25 22:32:20.886934684 +0900
@@ -20,7 +20,6 @@
 
 #include <stdio.h>
 #include <string.h>
-#include <glib/gstring.h>
 
 #include "mw_debug.h"
 #include "mw_util.h"
