--- a/Makefile.PL	2013-12-06 17:32:07.078910105 +0000
+++ b/Makefile.PL	2013-12-06 17:33:05.224179905 +0000
@@ -43,7 +43,7 @@
         $script .= <<"INSTALL";
 
 install_sax_pureperl :
-\t\@\$(PERL) -MXML::SAX -e "XML::SAX->add_parser(q(XML::SAX::PurePerl))->save_parsers()"
+\t\@\$(PERL) -I\$(DESTINSTALLSITELIB) -MXML::SAX -e "XML::SAX->add_parser(q(XML::SAX::PurePerl))->save_parsers()"
 
 INSTALL
 
