--- Makefile.PL.old	2013-04-18 01:59:35.176315107 +0000
+++ Makefile.PL	2013-04-18 02:00:32.436377014 +0000
@@ -25,18 +25,18 @@
     if ($@) {
         $write_ini_ok = 1;
     }
-    else {
-        my $dir = File::Basename::dirname($INC{'XML/SAX.pm'});
-        if (-e File::Spec->catfile($dir, 'SAX', 'ParserDetails.ini')) {
-            $write_ini_ok =
-                ExtUtils::MakeMaker::prompt(
-                    "Do you want XML::SAX to alter ParserDetails.ini?", "Y"
-                ) =~ /^y/i;
-        }
-        else {
-            $write_ini_ok = 1;
-        }
-    }
+#    else {
+#        my $dir = File::Basename::dirname($INC{'XML/SAX.pm'});
+#        if (-e File::Spec->catfile($dir, 'SAX', 'ParserDetails.ini')) {
+#            $write_ini_ok =
+#                ExtUtils::MakeMaker::prompt(
+#                    "Do you want XML::SAX to alter ParserDetails.ini?", "Y"
+#                ) =~ /^y/i;
+#        }
+#        else {
+#            $write_ini_ok = 1;
+#        }
+#    }
     
     if ($write_ini_ok) {
         $script =~ s/install :: (.*)$/install :: $1 install_sax_pureperl/m;
