diff -up gutenprint-5.2.7/src/cups/commandtocanon.c.build gutenprint-5.2.7/src/cups/commandtocanon.c
--- gutenprint-5.2.7/src/cups/commandtocanon.c.build	2011-06-07 13:32:37.372042304 +0100
+++ gutenprint-5.2.7/src/cups/commandtocanon.c	2011-06-07 13:32:41.270970750 +0100
@@ -13,6 +13,7 @@

 #include <cups/cups.h>
 #include <ctype.h>
+#include <stdio.h>
 #include <stdlib.h>


diff -up gutenprint-5.2.7/src/cups/commandtoepson.c.build gutenprint-5.2.7/src/cups/commandtoepson.c
--- gutenprint-5.2.7/src/cups/commandtoepson.c.build	2007-12-24 17:19:20.000000000 +0000
+++ gutenprint-5.2.7/src/cups/commandtoepson.c	2011-06-07 13:32:35.656073794 +0100
@@ -30,6 +30,7 @@

 #include <cups/cups.h>
 #include <ctype.h>
+#include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
