diff -rupN xf86-video-geode-2.11.14-orig/src/geode_driver.c xf86-video-geode-2.11.14/src/geode_driver.c
--- xf86-video-geode-2.11.14-orig/src/geode_driver.c	2013-03-09 15:16:56.957299445 +0100
+++ xf86-video-geode-2.11.14/src/geode_driver.c	2013-03-09 15:17:49.260088291 +0100
@@ -59,8 +59,6 @@
 /* Machine independent stuff */
 #include "mipointer.h"
 #include "micmap.h"
-/* All drivers implementing backing store need this */
-#include "mibstore.h"
 #include "vgaHW.h"
 #include "vbe.h"
 
diff -rupN xf86-video-geode-2.11.14-orig/src/gx_driver.c xf86-video-geode-2.11.14/src/gx_driver.c
--- xf86-video-geode-2.11.14-orig/src/gx_driver.c	2013-03-09 15:16:56.957299445 +0100
+++ xf86-video-geode-2.11.14/src/gx_driver.c	2013-03-09 15:18:18.461086994 +0100
@@ -1350,7 +1350,6 @@ GXScreenInit(SCREEN_INIT_ARGS_DECL)
     if (!pGeode->NoAccel)
         GXAccelInit(pScrn);
 
-    miInitializeBackingStore(pScrn);
     xf86SetBackingStore(pScrn);
 
     /* Set up the soft cursor */
diff -rupN xf86-video-geode-2.11.14-orig/src/lx_driver.c xf86-video-geode-2.11.14/src/lx_driver.c
--- xf86-video-geode-2.11.14-orig/src/lx_driver.c	2013-03-09 15:16:56.956299411 +0100
+++ xf86-video-geode-2.11.14/src/lx_driver.c	2013-03-09 15:18:26.861374290 +0100
@@ -932,7 +932,6 @@ LXScreenInit(SCREEN_INIT_ARGS_DECL)
     if (!pGeode->NoAccel)
         pGeode->NoAccel = LXExaInit(pScrn) ? FALSE : TRUE;
 
-    miInitializeBackingStore(pScrn);
     xf86SetBackingStore(pScrn);
 
     /* Set up the soft cursor */
