diff -urN wbar-2.3.4.orig/src/config/Functions.cc wbar-2.3.4/src/config/Functions.cc
--- wbar-2.3.4.orig/src/config/Functions.cc     2012-09-19 02:54:53.000000000 +0000
+++ wbar-2.3.4/src/config/Functions.cc  2018-06-03 18:48:52.415718498 +0000
@@ -11,7 +11,7 @@
 #include "Run.h"
 #include "OptParser.h"
 
-#define ICON_DEFAULT PIXMAPDIR"/"PACKAGE_NAME".png"
+#define ICON_DEFAULT PIXMAPDIR "/" PACKAGE_NAME".png"
 
 static const gchar *authors[] =
 {
@@ -304,7 +304,7 @@
         
         if (command.empty())
         {
-            command = PACKAGE_NAME" "DEFAULT_ARGV;
+            command = PACKAGE_NAME " " DEFAULT_ARGV;
         }
 
         if (argc <= 1 || tmpoptparser.isSet( OptParser::CONFIG ))
diff -urN wbar-2.3.4.orig/src/config/Run.cc wbar-2.3.4/src/config/Run.cc
--- wbar-2.3.4.orig/src/config/Run.cc   2012-03-02 14:11:44.000000000 +0000
+++ wbar-2.3.4/src/config/Run.cc        2018-06-03 18:44:47.880871828 +0000
@@ -136,7 +136,7 @@
 {
     if (Run::getPID() > 0)
     {
-        if (system ("killall "PACKAGE_NAME) != 0)
+        if (system ("killall " PACKAGE_NAME) != 0)
         {
             std::cout << _("Error kill program: ") << PACKAGE_NAME << std::endl;
         }
diff -urN wbar-2.3.4.orig/src/core/Main.cc wbar-2.3.4/src/core/Main.cc
--- wbar-2.3.4.orig/src/core/Main.cc    2012-09-18 02:26:41.000000000 +0000
+++ wbar-2.3.4/src/core/Main.cc 2018-06-03 18:45:59.789335064 +0000
@@ -81,7 +81,7 @@
 
         if (command.empty())
         {
-            command = PACKAGE_NAME" "DEFAULT_ARGV;
+            command = PACKAGE_NAME " " DEFAULT_ARGV;
         }
 
         if (argc <= 1 || tmpoptparser.isSet( OptParser::CONFIG ))


