# http://bugs.exim.org/show_bug.cgi?id=1164

diff -Nru exim-4.80.orig/OS/Makefile-Base exim-4.80/OS/Makefile-Base
--- exim-4.80.orig/OS/Makefile-Base	2012-05-31 15:48:44.513624313 +0200
+++ exim-4.80/OS/Makefile-Base	2012-05-31 16:23:02.159673311 +0200
@@ -457,7 +457,7 @@
 
 # Update Exim's version information and build the version object.
 
-version.h::
+version.h:
 	@../scripts/reversion
 
 cnumber.h: version.h
diff -Nru exim-4.80.orig/src/version.sh exim-4.80/src/version.sh
--- exim-4.80.orig/src/version.sh	2012-05-31 15:48:44.510291099 +0200
+++ exim-4.80/src/version.sh	2012-05-31 16:23:02.159673311 +0200
@@ -1,4 +1,4 @@
 # automatically generated file - see ../scripts/reversion
 EXIM_RELEASE_VERSION="4.80"
 EXIM_VARIANT_VERSION=""
-EXIM_COMPILE_NUMBER="1"
+EXIM_COMPILE_NUMBER="0"
