--- src/daemon/main.c.orig	2013-11-12 18:05:02.526028785 +0000
+++ src/daemon/main.c	2013-11-12 18:04:11.714663907 +0000
@@ -654,7 +654,7 @@
 
 #ifdef HAVE_GETUID
     if (getuid() == 0 && !conf->system_instance)
-        pa_log_warn(_("This program is not intended to be run as root (unless --system is specified)."));
+//        pa_log_warn(_("This program is not intended to be run as root (unless --system is specified)."));
 #ifndef HAVE_DBUS /* A similar, only a notice worthy check was done earlier, if D-Bus is enabled. */
     else if (getuid() != 0 && conf->system_instance) {
         pa_log(_("Root privileges required."));
