--- thunar/thunar-window.c.orig	2019-08-19 01:01:23.020750407 +0000
+++ thunar/thunar-window.c	2019-08-19 01:01:50.501948996 +0000
@@ -857,22 +857,6 @@
   exo_binding_new (G_OBJECT (window->spinner), "active",
                    G_OBJECT (window->spinner), "visible");
 
-  /* check if we need to add the root warning */
-  if (G_UNLIKELY (geteuid () == 0))
-    {
-      /* add the bar for the root warning */
-      infobar = gtk_info_bar_new ();
-      gtk_info_bar_set_message_type (GTK_INFO_BAR (infobar), GTK_MESSAGE_WARNING);
-      gtk_widget_set_hexpand (infobar, TRUE);
-      gtk_grid_attach (GTK_GRID (window->grid), infobar, 0, 2, 1, 1);
-      gtk_widget_show (infobar);
-
-      /* add the label with the root warning */
-      label = gtk_label_new (_("Warning, you are using the root account, you may harm your system."));
-      gtk_container_add (GTK_CONTAINER (gtk_info_bar_get_content_area (GTK_INFO_BAR (infobar))), label);
-      gtk_widget_show (label);
-    }
-
   window->paned = gtk_paned_new (GTK_ORIENTATION_HORIZONTAL);
   gtk_container_set_border_width (GTK_CONTAINER (window->paned), 0);
   gtk_widget_set_hexpand (window->paned, TRUE);
