diff -up openssl-1.1.1d/Configurations/unix-Makefile.tmpl.no-html openssl-1.1.1d/Configurations/unix-Makefile.tmpl
--- openssl-1.1.1d/Configurations/unix-Makefile.tmpl.no-html	2019-09-13 15:00:32.976774673 +0200
+++ openssl-1.1.1d/Configurations/unix-Makefile.tmpl	2019-09-13 15:02:22.283864321 +0200
@@ -544,7 +544,7 @@ install_sw: install_dev install_engines
 
 uninstall_sw: uninstall_runtime uninstall_engines uninstall_dev
 
-install_docs: install_man_docs install_html_docs
+install_docs: install_man_docs
 
 uninstall_docs: uninstall_man_docs uninstall_html_docs
 	$(RM) -r $(DESTDIR)$(DOCDIR)
