diff -up openssl-1.1.0h/crypto/rsa/rsa_gen.c.cc-reqs openssl-1.1.0h/crypto/rsa/rsa_gen.c
--- openssl-1.1.0h/crypto/rsa/rsa_gen.c.cc-reqs	2018-03-27 15:50:39.000000000 +0200
+++ openssl-1.1.0h/crypto/rsa/rsa_gen.c	2018-03-29 14:37:53.405048562 +0200
@@ -86,6 +86,12 @@ static int rsa_builtin_keygen(RSA *rsa,
     if (!rsa->iqmp && ((rsa->iqmp = BN_secure_new()) == NULL))
         goto err;
 
+    /* prepare minimum p and q difference */
+    if (!BN_one(r3))
+        goto err;
+    if (bitsp > 100 && !BN_lshift(r3, r3, bitsp - 100))
+        goto err;
+
     if (BN_copy(rsa->e, e_value) == NULL)
         goto err;
 
@@ -118,7 +124,9 @@ static int rsa_builtin_keygen(RSA *rsa,
         do {
             if (!BN_generate_prime_ex(rsa->q, bitsq, 0, NULL, NULL, cb))
                 goto err;
-        } while (BN_cmp(rsa->p, rsa->q) == 0);
+            if (!BN_sub(r2, rsa->q, rsa->p))
+                goto err;
+        } while (BN_ucmp(r2, r3) <= 0);
         if (!BN_sub(r2, rsa->q, BN_value_one()))
             goto err;
         ERR_set_mark();
