diff -up openssl-1.1.0-pre5/apps/s_client.c.xmpp openssl-1.1.0-pre5/apps/s_client.c
--- openssl-1.1.0-pre5/apps/s_client.c.xmpp	2016-07-20 14:53:39.008233281 +0200
+++ openssl-1.1.0-pre5/apps/s_client.c	2016-07-20 14:56:24.452962890 +0200
@@ -134,6 +134,8 @@
  * OTHERWISE.
  */
 
+/* for strcasestr */
+#define _GNU_SOURCE
 #include <ctype.h>
 #include <stdio.h>
 #include <stdlib.h>
@@ -2023,9 +2025,9 @@ int s_client_main(int argc, char **argv)
                        xmpphost ? xmpphost : host);
             seen = BIO_read(sbio, mbuf, BUFSIZZ);
             mbuf[seen] = 0;
-            while (!strstr
+            while (!strcasestr
                    (mbuf, "<starttls xmlns='urn:ietf:params:xml:ns:xmpp-tls'")
-                   && !strstr(mbuf,
+                   && !strcasestr(mbuf,
                               "<starttls xmlns=\"urn:ietf:params:xml:ns:xmpp-tls\""))
             {
                 seen = BIO_read(sbio, mbuf, BUFSIZZ);
