diff -ur a/gnome-dictionary/libgdict/gdict-context.h b/gnome-dictionary/libgdict/gdict-context.h
--- a/gnome-dictionary/libgdict/gdict-context.h	2009-09-21 15:54:41.000000000 +0000
+++ b/gnome-dictionary/libgdict/gdict-context.h	2013-12-27 17:44:33.503206742 +0000
@@ -21,6 +21,7 @@
 #define __GDICT_CONTEXT_H__
 
 #include <glib-object.h>
+#define G_CONST_RETURN const
 
 G_BEGIN_DECLS
 
diff -ur a/gnome-dictionary/libgdict/gdict-defbox.c b/gnome-dictionary/libgdict/gdict-defbox.c
--- a/gnome-dictionary/libgdict/gdict-defbox.c	2010-07-16 15:18:51.000000000 +0000
+++ b/gnome-dictionary/libgdict/gdict-defbox.c	2013-12-27 17:46:03.087086789 +0000
@@ -393,7 +393,7 @@
 	/* If string contains prefix, check that prefix is not followed
 	 * by a unicode mark symbol, e.g. that trailing 'a' in prefix
 	 * is not part of two-char a-with-hat symbol in string. */
-	return type != G_UNICODE_COMBINING_MARK &&
+	return type != G_UNICODE_SPACING_MARK &&
 		type != G_UNICODE_ENCLOSING_MARK &&
 		type != G_UNICODE_NON_SPACING_MARK;
 }
diff -ur a/gsearchtool/libgnomeui-deprecated/Makefile.in b/gsearchtool/libgnomeui-deprecated/Makefile.in
--- a/gsearchtool/libgnomeui-deprecated/Makefile.in	2010-09-27 22:50:42.000000000 +0000
+++ b/gsearchtool/libgnomeui-deprecated/Makefile.in	2013-12-27 17:58:01.629103778 +0000
@@ -278,9 +278,6 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-INCLUDES = -DGTK_DISABLE_DEPRECATED \
-           -DGDK_DISABLE_DEPRECATED \
-           -DG_DISABLE_DEPRECATED
 
 noinst_LTLIBRARIES = libgnomeui-deprecated.la
 libgnomeui_deprecated_la_LIBADD = $(GNOME_UTILS_LIBS)
