From Rawhide. Show Virtual Name in netstat.

diff -Naur net-tools-1.60.orig/lib/interface.c net-tools-1.60/lib/interface.c
--- net-tools-1.60.orig/lib/interface.c	2001-02-10 19:31:15.000000000 +0000
+++ net-tools-1.60/lib/interface.c	2003-09-12 22:50:35.000000000 +0000
@@ -579,7 +579,7 @@
 
 void ife_print_short(struct interface *ptr)
 {
-    printf("%-5.5s ", ptr->name);
+    printf("%-9.8s ", ptr->name);
     printf("%5d %3d", ptr->mtu, ptr->metric);
     /* If needed, display the interface statistics. */
     if (ptr->statistics_valid) {
diff -Naur net-tools-1.60.orig/netstat.c net-tools-1.60/netstat.c
--- net-tools-1.60.orig/netstat.c	2003-09-12 22:43:54.000000000 +0000
+++ net-tools-1.60/netstat.c	2003-09-12 22:50:35.000000000 +0000
@@ -1449,7 +1449,7 @@
     }
     if (flag_exp < 2) {
 	ife_short = 1;
-	printf(_("Iface   MTU Met   RX-OK RX-ERR RX-DRP RX-OVR   TX-OK TX-ERR TX-DRP TX-OVR Flg\n"));
+	printf(_("Iface     MTU Met   RX-OK RX-ERR RX-DRP RX-OVR   TX-OK TX-ERR TX-DRP TX-OVR Flg\n"));
     }
 
     if (for_all_interfaces(do_if_print, &flag_all) < 0) {
