diff -Nru net-tools-1.60/mii-tool.c net-tools-1.60-new/mii-tool.c
--- net-tools-1.60/mii-tool.c	2000-05-21 16:31:17.000000000 +0200
+++ net-tools-1.60-new/mii-tool.c	2003-11-08 02:11:31.000000000 +0100
@@ -379,17 +379,17 @@
 /*--------------------------------------------------------------------*/
 
 const char *usage =
-"usage: %s [-VvRrwl] [-A media,... | -F media] [interface ...]
-       -V, --version               display version information
-       -v, --verbose               more verbose output
-       -R, --reset                 reset MII to poweron state
-       -r, --restart               restart autonegotiation
-       -w, --watch                 monitor for link status changes
-       -l, --log                   with -w, write events to syslog
-       -A, --advertise=media,...   advertise only specified media
-       -F, --force=media           force specified media technology
-media: 100baseT4, 100baseTx-FD, 100baseTx-HD, 10baseT-FD, 10baseT-HD,
-       (to advertise both HD and FD) 100baseTx, 10baseT\n";
+"usage: %s [-VvRrwl] [-A media,... | -F media] [interface ...]\n"
+"       -V, --version               display version information\n"
+"       -v, --verbose               more verbose output\n"
+"       -R, --reset                 reset MII to poweron state\n"
+"       -r, --restart               restart autonegotiation\n"
+"       -w, --watch                 monitor for link status changes\n"
+"       -l, --log                   with -w, write events to syslog\n"
+"       -A, --advertise=media,...   advertise only specified media\n"
+"       -F, --force=media           force specified media technology\n"
+"media: 100baseT4, 100baseTx-FD, 100baseTx-HD, 10baseT-FD, 10baseT-HD,\n"
+"       (to advertise both HD and FD) 100baseTx, 10baseT\n";
 
 int main(int argc, char **argv)
 {
diff -ru net-tools-1.60/lib/x25_sr.c net-tools-1.60-new/lib/x25_sr.c
--- net-tools-1.60/lib/x25_sr.c	2000-05-20 15:38:10.000000000 +0200
+++ net-tools-1.60-new/lib/x25_sr.c	2004-04-30 23:37:10.536499600 +0200
@@ -77,7 +77,7 @@
   rt.sigdigits=sigdigits;
 
   /* x25_route_struct.address isn't type struct sockaddr_x25, Why? */
-  memcpy(&rt.address, &sx25.sx25_addr, sizeof(x25_address));
+  memcpy(&rt.address, &sx25.sx25_addr, sizeof(struct x25_address));
 
   while (*args) {
 	if (!strcmp(*args,"device") || !strcmp(*args,"dev")) {
diff -ru net-tools-1.60/lib/inet6_sr.c net-tools-1.60-new/lib/inet6_sr.c
--- net-tools-1.60/lib/inet6_sr.c	2000-05-22 23:18:37.000000000 +0200
+++ net-tools-1.60-new/lib/inet6_sr.c	2004-04-30 23:48:24.146095376 +0200
@@ -32,6 +32,7 @@
 #include "net-support.h"
 #include "pathnames.h"
 #include "intl.h"
+#include "util.h"
 #include "net-features.h"
 
 
diff -ru net-tools-1.60/lib/inet_sr.c net-tools-1.60-new/lib/inet_sr.c
--- net-tools-1.60/lib/inet_sr.c	2000-02-20 22:46:45.000000000 +0100
+++ net-tools-1.60-new/lib/inet_sr.c	2004-05-01 00:01:21.358941064 +0200
@@ -105,6 +105,7 @@
     case 2:
        isnet = 0; break;
     default:
+       break;
     }
 
     /* Fill in the other fields. */
diff -ru net-tools-1.60/hostname.c net-tools-1.60-new/hostname.c
--- net-tools-1.60/hostname.c	2001-04-08 19:04:23.000000000 +0200
+++ net-tools-1.60-new/hostname.c	2004-05-01 00:03:32.373023896 +0200
@@ -31,6 +31,7 @@
  *              your option) any later version.
  */
 #include <stdio.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <getopt.h>
 #include <string.h>
@@ -78,6 +79,7 @@
             fprintf(stderr, _("%s: name too long\n"), program_name);
             break;
         default:
+	    break;
         }
 	exit(1);
     }
@@ -98,6 +100,7 @@
 	    fprintf(stderr, _("%s: name too long\n"), program_name);
 	    break;
 	default:
+	    break;
 	}
 	exit(1);
     };
@@ -117,6 +120,7 @@
 	    fprintf(stderr, _("%s: name too long\n"), program_name);
 	    break;
 	default:
+	    break;
 	}
 	exit(1);
     };
@@ -174,6 +178,7 @@
 	printf("%s\n", hp->h_name);
 	break;
     default:
+	break;
     }
 }
 
@@ -330,7 +335,7 @@
 	case 'h':
 	default:
 	    usage();
-
+	    break;
 	};
 
 
