--- glibc-2.18.orig/sysdeps/i386/i686/Makefile	2013-12-04 22:31:36.018800151 +0000
+++ glibc-2.18/sysdeps/i386/i686/Makefile	2013-12-04 22:33:02.126013001 +0000
@@ -10,16 +10,18 @@
 sysdep_routines += cacheinfo
 endif
 
-ifeq (yes,$(config-asflags-i686))
-CFLAGS-.o += -Wa,-mtune=i686
-CFLAGS-.os += -Wa,-mtune=i686
-CFLAGS-.op += -Wa,-mtune=i686
-CFLAGS-.og += -Wa,-mtune=i686
-CFLAGS-.oS += -Wa,-mtune=i686
-
-ASFLAGS-.o += -Wa,-mtune=i686
-ASFLAGS-.os += -Wa,-mtune=i686
-ASFLAGS-.op += -Wa,-mtune=i686
-ASFLAGS-.og += -Wa,-mtune=i686
-ASFLAGS-.oS += -Wa,-mtune=i686
-endif
+# Disabled for Fedora for Geode LX support
+# see https://bugzilla.redhat.com/show_bug.cgi?id=579838
+#ifeq (yes,$(config-asflags-i686))
+#CFLAGS-.o += -Wa,-mtune=i686
+#CFLAGS-.os += -Wa,-mtune=i686
+#CFLAGS-.op += -Wa,-mtune=i686
+#CFLAGS-.og += -Wa,-mtune=i686
+#CFLAGS-.oS += -Wa,-mtune=i686
+#
+#ASFLAGS-.o += -Wa,-mtune=i686
+#ASFLAGS-.os += -Wa,-mtune=i686
+#ASFLAGS-.op += -Wa,-mtune=i686
+#ASFLAGS-.og += -Wa,-mtune=i686
+#ASFLAGS-.oS += -Wa,-mtune=i686
+#endif
