--- term-utils/agetty.c.orig	2019-01-31 20:35:02.160238682 +0000
+++ term-utils/agetty.c	2019-01-31 20:35:21.550382284 +0000
@@ -486,7 +486,6 @@
 			/* Autologin prompt */
 			eval_issue_file(&issue, &options, &termios);
 			do_prompt(&issue, &options, &termios);
-			printf(_("%s%s (automatic login)\n"), LOGIN, options.autolog);
 		} else {
 			/* Read the login name. */
 			debug("reading login name\n");
