--- BaseTools/Conf/tools_def.template.orig	2014-10-22 19:01:59.000000000 +0000
+++ BaseTools/Conf/tools_def.template	2014-10-22 19:03:47.601309731 +0000
@@ -138,13 +138,14 @@
 
 #
 # Option 1: Hard coded full path to compiler suite
-DEFINE UNIXGCC_IA32_PETOOLS_PREFIX = /opt/tiano/i386-tiano-pe/i386-tiano-pe/bin/
-DEFINE UNIXGCC_X64_PETOOLS_PREFIX  = /opt/tiano/x86_64-pc-mingw64/x86_64-pc-mingw64/bin/
-DEFINE UNIXGCC_IPF_PETOOLS_PREFIX  = /opt/tiano/ia64-pc-elf/ia64-pc-elf/bin/
+#DEFINE UNIXGCC_IA32_PETOOLS_PREFIX = /opt/tiano/i386-tiano-pe/i386-tiano-pe/bin/
+#DEFINE UNIXGCC_X64_PETOOLS_PREFIX  = /opt/tiano/x86_64-pc-mingw64/x86_64-pc-mingw64/bin/
+#DEFINE UNIXGCC_IPF_PETOOLS_PREFIX  = /opt/tiano/ia64-pc-elf/ia64-pc-elf/bin/
 #
 # Option 2: Use an environment variable
-#DEFINE UNIXGCC_IA32_PETOOLS_PREFIX = ENV(IA32_PETOOLS_PREFIX)
-#DEFINE UNIXGCC_X64_PETOOLS_PREFIX  = ENV(X64_PETOOLS_PREFIX)
+DEFINE UNIXGCC_IA32_PETOOLS_PREFIX = ENV(IA32_PETOOLS_PREFIX)
+DEFINE UNIXGCC_X64_PETOOLS_PREFIX  = ENV(X64_PETOOLS_PREFIX)
+DEFINE UNIXGCC_IPF_PETOOLS_PREFIX  = ENV(X64_PETOOLS_PREFIX
 #
 # Option 3: Install the compiler suite into your default paths
 #DEFINE UNIXGCC_IA32_PETOOLS_PREFIX = i386-pc-mingw32-
--- BaseTools/Conf/tools_def.template.orig	2014-10-22 20:25:59.591671082 +0000
+++ BaseTools/Conf/tools_def.template	2014-10-22 20:27:16.044908789 +0000
@@ -145,7 +145,7 @@
 # Option 2: Use an environment variable
 DEFINE UNIXGCC_IA32_PETOOLS_PREFIX = ENV(IA32_PETOOLS_PREFIX)
 DEFINE UNIXGCC_X64_PETOOLS_PREFIX  = ENV(X64_PETOOLS_PREFIX)
-DEFINE UNIXGCC_IPF_PETOOLS_PREFIX  = ENV(X64_PETOOLS_PREFIX
+DEFINE UNIXGCC_IPF_PETOOLS_PREFIX  = ENV(X64_PETOOLS_PREFIX)
 #
 # Option 3: Install the compiler suite into your default paths
 #DEFINE UNIXGCC_IA32_PETOOLS_PREFIX = i386-pc-mingw32-
@@ -180,8 +180,8 @@
 DEFINE GCC47_IA32_PREFIX       = /usr/bin/
 DEFINE GCC47_X64_PREFIX        = /usr/bin/
 
-DEFINE GCC48_IA32_PREFIX       = /usr/bin/
-DEFINE GCC48_X64_PREFIX        = /usr/bin/
+DEFINE GCC48_IA32_PREFIX       = ENV(IA32_PETOOLS_PREFIX)
+DEFINE GCC48_X64_PREFIX        = ENV(X64_PETOOLS_PREFIX)
 
 DEFINE GCC49_IA32_PREFIX       = /usr/bin/
 DEFINE GCC49_X64_PREFIX        = /usr/bin/
--- BaseTools/Conf/tools_def.template.orig	2014-10-23 13:02:38.908549143 +0000
+++ BaseTools/Conf/tools_def.template	2014-10-23 13:07:20.976891338 +0000
@@ -134,7 +134,7 @@
 DEFINE EBC_BIN          = C:\Program Files\Intel\EBC\Bin
 DEFINE EBC_BINx86       = C:\Program Files (x86)\Intel\EBC\Bin
 
-DEFINE ELFGCC_BIN       = /usr/bin
+DEFINE ELFGCC_BIN       = ENV(X64_PETOOLS_PREFIX)
 
 #
 # Option 1: Hard coded full path to compiler suite
@@ -166,25 +166,25 @@
 DEFINE CYGWIN_BINX64           = c:/cygwin/opt/tiano/x86_64-pc-mingw64/x86_64-pc-mingw64/bin/
 DEFINE CYGWIN_BINIPF           = c:/cygwin/opt/tiano/gcc/ipf/bin/ia64-pc-elf-
 
-DEFINE GCC44_IA32_PREFIX       = /usr/bin/
+DEFINE GCC44_IA32_PREFIX       = ENV(X64_PETOOLS_PREFIX)
 #DEFINE GCC44_IA32_PREFIX       = ENV(HOME)/programs/gcc/4.4/ia32/
-DEFINE GCC44_X64_PREFIX        = /usr/bin/
+DEFINE GCC44_X64_PREFIX        = ENV(X64_PETOOLS_PREFIX)
 #DEFINE GCC44_X64_PREFIX       = ENV(HOME)/programs/gcc/4.4/x64/
 
-DEFINE GCC45_IA32_PREFIX       = /usr/bin/
-DEFINE GCC45_X64_PREFIX        = /usr/bin/
+DEFINE GCC45_IA32_PREFIX       = ENV(X64_PETOOLS_PREFIX)
+DEFINE GCC45_X64_PREFIX        = ENV(X64_PETOOLS_PREFIX)
 
-DEFINE GCC46_IA32_PREFIX       = /usr/bin/
-DEFINE GCC46_X64_PREFIX        = /usr/bin/
+DEFINE GCC46_IA32_PREFIX       = ENV(X64_PETOOLS_PREFIX)
+DEFINE GCC46_X64_PREFIX        = ENV(X64_PETOOLS_PREFIX)
 
-DEFINE GCC47_IA32_PREFIX       = /usr/bin/
-DEFINE GCC47_X64_PREFIX        = /usr/bin/
+DEFINE GCC47_IA32_PREFIX       = ENV(X64_PETOOLS_PREFIX)
+DEFINE GCC47_X64_PREFIX        = ENV(X64_PETOOLS_PREFIX)
 
-DEFINE GCC48_IA32_PREFIX       = ENV(IA32_PETOOLS_PREFIX)
+DEFINE GCC48_IA32_PREFIX       = ENV(X64_PETOOLS_PREFIX)
 DEFINE GCC48_X64_PREFIX        = ENV(X64_PETOOLS_PREFIX)
 
-DEFINE GCC49_IA32_PREFIX       = /usr/bin/
-DEFINE GCC49_X64_PREFIX        = /usr/bin/
+DEFINE GCC49_IA32_PREFIX       = ENV(X64_PETOOLS_PREFIX)
+DEFINE GCC49_X64_PREFIX        = ENV(X64_PETOOLS_PREFIX)
 
 DEFINE UNIX_IASL_BIN           = ENV(IASL_PREFIX)iasl
 DEFINE WIN_ASL_BIN_DIR         = C:\ASL
--- BaseTools/Conf/tools_def.template.orig	2014-10-23 13:20:52.056384100 +0000
+++ BaseTools/Conf/tools_def.template	2014-10-23 13:28:43.213004540 +0000
@@ -134,7 +134,7 @@
 DEFINE EBC_BIN          = C:\Program Files\Intel\EBC\Bin
 DEFINE EBC_BINx86       = C:\Program Files (x86)\Intel\EBC\Bin
 
-DEFINE ELFGCC_BIN       = ENV(X64_PETOOLS_PREFIX)
+DEFINE ELFGCC_BIN       = /usr/bin
 
 #
 # Option 1: Hard coded full path to compiler suite
@@ -6320,16 +6320,16 @@
 # X64 definitions - still a work in progress. This tool chain does not produce
 # the correct ABI, it is just used to compile the code....
 ##################
-*_XCODE32_X64_CC_PATH       = gcc
+*_XCODE32_X64_CC_PATH       = /opt/cross/x-tools/x86_64-thinstation-linux-gnu/bin/x86_64-thinstation-linux-gnu-gcc
 *_XCODE32_X64_SLINK_PATH    = libtool
-*_XCODE32_X64_DLINK_PATH    = ld
-*_XCODE32_X64_ASM_PATH      = as
-*_XCODE32_X64_PP_PATH       = gcc
-*_XCODE32_X64_VFRPP_PATH    = gcc
+*_XCODE32_X64_DLINK_PATH    = /opt/cross/x-tools/x86_64-thinstation-linux-gnu/bin/x86_64-thinstation-linux-gnu-ld
+*_XCODE32_X64_ASM_PATH      = /opt/cross/x-tools/x86_64-thinstation-linux-gnu/bin/x86_64-thinstation-linux-gnu-as
+*_XCODE32_X64_PP_PATH       = /opt/cross/x-tools/x86_64-thinstation-linux-gnu/bin/x86_64-thinstation-linux-gnu-gcc
+*_XCODE32_X64_VFRPP_PATH    = /opt/cross/x-tools/x86_64-thinstation-linux-gnu/bin/x86_64-thinstation-linux-gnu-gcc
 *_XCODE32_X64_ASL_PATH     = iasl
-*_XCODE32_X64_ASLCC_PATH   = gcc
-*_XCODE32_X64_ASLPP_PATH   = gcc
-*_XCODE32_X64_ASLDLINK_PATH  = ld
+*_XCODE32_X64_ASLCC_PATH   = /opt/cross/x-tools/x86_64-thinstation-linux-gnu/bin/x86_64-thinstation-linux-gnu-gcc
+*_XCODE32_X64_ASLPP_PATH   = /opt/cross/x-tools/x86_64-thinstation-linux-gnu/bin/x86_64-thinstation-linux-gnu-gcc
+*_XCODE32_X64_ASLDLINK_PATH  = /opt/cross/x-tools/x86_64-thinstation-linux-gnu/bin/x86_64-thinstation-linux-gnu-ld
 
 *_XCODE32_X64_DLINK_FLAGS      = -arch x86_64 -u _$(IMAGE_ENTRY_POINT) -e _$(IMAGE_ENTRY_POINT) -preload -segalign 0x20  -pie -seg1addr 0x240 -read_only_relocs suppress -map $(DEST_DIR_DEBUG)/$(BASE_NAME).map
 *_XCODE32_X64_SLINK_FLAGS      = -static -o
