--- .footprint.old	2011-11-28 14:06:46.394219605 -0600
+++ .footprint	2011-11-28 14:06:32.529392941 -0600
@@ -1,4 +1,7 @@
 drwxr-xr-x	root/root	etc/
+drwxr-xr-x	root/root	etc/dbus-1/
+drwxr-xr-x	root/root	etc/dbus-1/system.d/
+-rw-r--r--	root/root	etc/dbus-1/system.d/wpa_supplicant.conf
 -rw-------	root/root	etc/wpa.conf
 drwxr-xr-x	root/root	usr/
 drwxr-xr-x	root/root	usr/man/
@@ -13,3 +16,7 @@
 -rwxr-xr-x	root/root	usr/sbin/wpa_cli
 -rwxr-xr-x	root/root	usr/sbin/wpa_passphrase
 -rwxr-xr-x	root/root	usr/sbin/wpa_supplicant
+drwxr-xr-x	root/root	usr/share/
+drwxr-xr-x	root/root	usr/share/dbus-1/
+drwxr-xr-x	root/root	usr/share/dbus-1/system-services/
+-rw-r--r--	root/root	usr/share/dbus-1/system-services/fi.w1.wpa_supplicant1.service
--- Pkgfile.old	2011-11-28 14:06:50.788164675 -0600
+++ Pkgfile	2011-11-29 11:51:25.788587165 -0600
@@ -12,11 +12,16 @@
     cd $name-$version/$name
     cp defconfig .config
     echo "CONFIG_READLINE=y" >> .config
+    echo "CONFIG_CTRL_IFACE_DBUS_NEW=y" >> .config
+    echo "CONFIG_DRIVER_NL80211=y" >> .config
     make
     install -d $PKG/{usr/sbin,usr/man/man{8,5},etc}
+    install -d $PKG/{etc/dbus-1/system.d,usr/share/dbus-1/system-services}
     install wpa_{cli,passphrase,supplicant} $PKG/usr/sbin
     install doc/docbook/wpa_{background,cli,passphrase,supplicant}.8 $PKG/usr/man/man8
     install doc/docbook/wpa_supplicant.conf.5 $PKG/usr/man/man5
+    install -m 644 dbus/dbus-wpa_supplicant.conf $PKG/etc/dbus-1/system.d/wpa_supplicant.conf
+    install -m 644 dbus/fi.w1.wpa_supplicant1.service $PKG/usr/share/dbus-1/system-services
     echo -e "ctrl_interface=/var/run/wpa_supplicant\n" > $PKG/etc/wpa.conf
     chmod 0600 $PKG/etc/wpa.conf
 }
