--- vncviewer/CMakeLists.txt.old	2013-04-10 22:18:59.032714776 +0000
+++ vncviewer/CMakeLists.txt	2013-04-10 22:19:29.733744299 +0000
@@ -38,7 +38,7 @@
 endif()
 
 if(USE_INCLUDED_FLTK)
-  target_link_libraries(vncviewer fltk_static)
+  target_link_libraries(vncviewer fltk_static fontconfig)
 endif()
 target_link_libraries(vncviewer rfb network rdr os Xregion ${FLTK_LIBRARIES} ${GETTEXT_LIBRARIES})
 
