diff -ur tigervnc-1.2.0-old/common/rdr/TLSInStream.cxx tigervnc-1.2.0/common/rdr/TLSInStream.cxx
--- tigervnc-1.2.0-old/common/rdr/TLSInStream.cxx	2013-04-21 00:37:43.320168069 +0000
+++ tigervnc-1.2.0/common/rdr/TLSInStream.cxx	2013-04-21 00:38:10.878197854 +0000
@@ -27,9 +27,7 @@
 #include <rdr/TLSInStream.h>
 #include <errno.h>
 
-#ifdef HAVE_OLD_GNUTLS
 #define gnutls_transport_set_global_errno(A) do { errno = (A); } while(0)
-#endif
 
 #ifdef HAVE_GNUTLS 
 using namespace rdr;
diff -ur tigervnc-1.2.0-old/common/rdr/TLSOutStream.cxx tigervnc-1.2.0/common/rdr/TLSOutStream.cxx
--- tigervnc-1.2.0-old/common/rdr/TLSOutStream.cxx	2013-04-21 00:37:43.320168069 +0000
+++ tigervnc-1.2.0/common/rdr/TLSOutStream.cxx	2013-04-21 00:38:33.534222358 +0000
@@ -27,9 +27,7 @@
 #include <rdr/TLSOutStream.h>
 #include <errno.h>
 
-#ifdef HAVE_OLD_GNUTLS
 #define gnutls_transport_set_global_errno(A) do { errno = (A); } while(0)
-#endif
 
 #ifdef HAVE_GNUTLS
 using namespace rdr;
