--- libnl-1.1/lib/Makefile.orig	2008-01-15 00:48:45.000000000 +0900
+++ libnl-1.1/lib/Makefile	2010-02-19 10:35:51.314565974 +0900
@@ -34,6 +34,7 @@
 OBJ      := $(CIN:%.c=%.o)
 CFLAGS   += -fPIC
 OUT_SLIB := $(PACKAGE_NAME).so.$(PACKAGE_VERSION)
+OUT_AR   := $(PACKAGE_NAME).a
 LN_SLIB  := $(PACKAGE_NAME).so
 LN1_SLIB := $(LN_SLIB).1
 
@@ -44,7 +45,8 @@
 
 all:
 	@echo "  MAKE $(OUT_SLIB)"; \
-	$(MAKE) $(OUT_SLIB)
+	$(MAKE) $(OUT_SLIB); \
+	$(MAKE) $(OUT_AR)
 
 $(OUT_SLIB): ../Makefile.opts $(OBJ)
 	@echo "  LD $(OUT_SLIB)"; \
@@ -54,6 +56,10 @@
 	@echo "  LN $(LN1_SLIB) $(LN_SLIB)"; \
 	rm -f $(LN_SLIB) ; $(LN) -s $(LN1_SLIB) $(LN_SLIB)
 
+$(OUT_AR):  ../Makefile.opts $(OBJ)
+	@echo "  AR $(OUT_AR)"; \
+	$(AR) crus $(OUT_AR) $(OBJ)
+
 clean:
 	@echo "  CLEAN lib"; \
 	$(RM) -f $(OBJ) $(OUT_SLIB) $(LN_SLIB) $(LN1_SLIB); \
@@ -70,6 +76,7 @@
 	$(LN) -s $(OUT_SLIB) $(DESTDIR)$(libdir)/$(LN1_SLIB)
 	rm -f $(DESTDIR)$(libdir)/$(LN_SLIB)
 	$(LN) -s $(LN1_SLIB) $(DESTDIR)$(libdir)/$(LN_SLIB)
+	install -m 0644 $(OUT_AR) $(DESTDIR)$(libdir)
 
 $(DEPS): ../Makefile.opts
 
