--- a/configure	2014-11-10 20:40:53.997144224 +0000
+++ b/configure	2014-11-10 20:38:58.165143325 +0000
@@ -5137,6 +5137,8 @@
     '') ac_prog_version="v. ?.??, bad"; ac_verc_fail=yes;;
     3.79* | 3.[89]*)
        ac_prog_version="$ac_prog_version, ok"; ac_verc_fail=no;;
+    4.*)
+       ac_prog_version="$ac_prog_version, ok"; ac_verc_fail=no;;
     *) ac_prog_version="$ac_prog_version, bad"; ac_verc_fail=yes;;
 
   esac
