diff -ur qt-x11-free-3.3.8b/configure qt-x11-free-3.3.8b-gcc4-buildkey/configure
--- qt-x11-free-3.3.8b/configure	2008-01-15 20:09:15.000000000 +0100
+++ qt-x11-free-3.3.8b-gcc4-buildkey/configure	2008-02-18 02:41:01.000000000 +0100
@@ -3081,16 +3081,16 @@
 g++*)
     # GNU C++
     QMAKE_CONF_COMPILER=`grep "QMAKE_CXX[^_A-Z0-9a-z]" $QMAKESPEC/qmake.conf | sed "s,.* *= *\(.*\)$,\1,"`
-    COMPILER_VERSION=`${QMAKE_CONF_COMPILER} --version 2>/dev/null`
+    COMPILER_VERSION=`${QMAKE_CONF_COMPILER} --version 2>/dev/null | sed 's,^[^0-9]*,,g'`
     case "$COMPILER_VERSION" in
-    *2.95.*)
+    2.95.*)
 	COMPILER_VERSION="2.95.*"
 	;;
-    *3.*)
+    3.*)
 	COMPILER_VERSION="3.*"
 	;;
-    *4.*)
-	COMPILER_VERSION="4"
+    4.*)
+	COMPILER_VERSION="4.*"
 	;;
     *)
 	;;
