--- plugins/xrandr/gsd-xrandr-manager.c.orig	2014-04-18 18:53:41.752297210 +0000
+++ plugins/xrandr/gsd-xrandr-manager.c	2014-04-18 18:54:48.247889016 +0000
@@ -1094,8 +1094,7 @@
         else
                 notification = notify_notification_new (primary_text,
                                                         error_to_display ? error_to_display->message : secondary_text,
-                                                        GSD_XRANDR_ICON_NAME,
-                                                        NULL);
+                                                        GSD_XRANDR_ICON_NAME);
 
         notify_notification_show (notification, NULL); /* NULL-GError */
 #else
--- plugins/a11y-keyboard/gsd-a11y-keyboard-manager.c.orig	2014-04-18 18:58:08.834696152 +0000
+++ plugins/a11y-keyboard/gsd-a11y-keyboard-manager.c	2014-04-18 18:58:31.738587364 +0000
@@ -608,8 +608,7 @@
         gsd_a11y_keyboard_manager_ensure_status_icon (manager);
         manager->priv->notification = notify_notification_new (title,
                                                                message,
-                                                               "preferences-desktop-accessibility",
-                                                               NULL);
+                                                               "preferences-desktop-accessibility");
         notify_notification_attach_to_status_icon (manager->priv->notification, manager->priv->status_icon);
         notify_notification_set_timeout (manager->priv->notification, NOTIFICATION_TIMEOUT * 1000);
 
--- plugins/a11y-keyboard/gsd-a11y-keyboard-manager.c.orig	2014-04-18 19:17:33.095990067 +0000
+++ plugins/a11y-keyboard/gsd-a11y-keyboard-manager.c	2014-04-18 19:17:58.316973026 +0000
@@ -746,8 +746,7 @@
         gsd_a11y_keyboard_manager_ensure_status_icon (manager);
         manager->priv->notification = notify_notification_new (title,
                                                                message,
-                                                               "preferences-desktop-accessibility",
-                                                               NULL);
+                                                               "preferences-desktop-accessibility");
         notify_notification_attach_to_status_icon (manager->priv->notification, manager->priv->status_icon);
         notify_notification_set_timeout (manager->priv->notification, NOTIFICATION_TIMEOUT * 1000);
 
