diff -Nru httpd-2.2.6.orig/docs/conf/extra/httpd-ssl.conf.in httpd-2.2.6/docs/conf/extra/httpd-ssl.conf.in
--- httpd-2.2.6.orig/docs/conf/extra/httpd-ssl.conf.in	2007-09-07 16:24:03.000000000 +0200
+++ httpd-2.2.6/docs/conf/extra/httpd-ssl.conf.in	2007-09-07 16:27:15.000000000 +0200
@@ -96,7 +96,7 @@
 #   in mind that if you have both an RSA and a DSA certificate you
 #   can configure both in parallel (to also allow the use of DSA
 #   ciphers, etc.)
-SSLCertificateFile "@exp_sysconfdir@/server.crt"
+SSLCertificateFile "/etc/ssl/certs/apache.crt"
 #SSLCertificateFile "@exp_sysconfdir@/server-dsa.crt"
 
 #   Server Private Key:
@@ -104,7 +104,7 @@
 #   directive to point at the key file.  Keep in mind that if
 #   you've both a RSA and a DSA private key you can configure
 #   both in parallel (to also allow the use of DSA ciphers, etc.)
-SSLCertificateKeyFile "@exp_sysconfdir@/server.key"
+SSLCertificateKeyFile "/etc/ssl/keys/apache.key"
 #SSLCertificateKeyFile "@exp_sysconfdir@/server-dsa.key"
 
 #   Server Certificate Chain:
