--- xfce4-session/xfsm-logout-dialog.c.orig	2021-06-02 00:28:57.244411069 +0000
+++ xfce4-session/xfsm-logout-dialog.c	2021-06-02 00:44:25.436958617 +0000
@@ -184,7 +184,7 @@
   gtk_widget_show (main_vbox);
 
   /* label showing the users' name */
-  username = g_get_real_name ();
+/*  username = g_get_real_name ();
   if (username == NULL
       || *username == '\0'
       || strcmp ("Unknown", username) == 0)
@@ -235,7 +235,7 @@
   /**
    * Logout
    **/
-  button = xfsm_logout_dialog_button (_("_Log Out"), "xfsm-logout",
+  button = xfsm_logout_dialog_button (_("_Reload"), "reload",
                                       "system-log-out", NULL,
                                       XFSM_SHUTDOWN_LOGOUT, dialog);
 
@@ -417,14 +417,14 @@
   /**
    * Save session
    **/
-  if (xfsm_shutdown_can_save_session (dialog->shutdown)
-      && !xfconf_channel_get_bool (channel, "/general/AutoSave", FALSE))
-    {
-      dialog->save_session = gtk_check_button_new_with_mnemonic (_("_Save session for future logins"));
-      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (dialog->save_session), save_session);
-      gtk_box_pack_start (GTK_BOX (vbox), dialog->save_session, FALSE, TRUE, BORDER);
-      gtk_widget_show (dialog->save_session);
-    }
+//  if (xfsm_shutdown_can_save_session (dialog->shutdown)
+//      && !xfconf_channel_get_bool (channel, "/general/AutoSave", FALSE))
+//    {
+//      dialog->save_session = gtk_check_button_new_with_mnemonic (_("_Save session for future logins"));
+//      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (dialog->save_session), save_session);
+//      gtk_box_pack_start (GTK_BOX (vbox), dialog->save_session, FALSE, TRUE, BORDER);
+//      gtk_widget_show (dialog->save_session);
+//    }
 
   attrs = pango_attr_list_new ();
   pango_attr_list_insert (attrs, pango_attr_weight_new (PANGO_WEIGHT_BOLD));
