--- src/atasks.cc.orig	2006-01-22 14:38:02.000000000 +0100
+++ src/atasks.cc	2006-01-22 14:38:17.000000000 +0100
@@ -212,8 +212,8 @@
 
         if (small != null) {
             int const y((height() - 3 - small->height() - 
-                         ((wmLook == lookMetal) ? 1 : 0)) / 2);
-            g.drawImage(small, p + 1, p + 1 + y);
+                         ((wmLook == lookMetal) ? 2 : 0)) / 2);
+            g.drawImage(small, p + 3, p + 1 + y);
         }
     }
 #endif
@@ -232,7 +232,7 @@
             int iconSize = 0;
 #ifndef LITE
             if (taskBarShowWindowIcons)
-                iconSize = YIcon::smallSize();
+                iconSize = YIcon::smallSize() + 3;
 #endif
             int const tx = 3 + iconSize;
             int const ty = max(2,
