diff -Nur xfce4-battery-plugin-0.5.1.orig/panel-plugin/libapm.h xfce4-battery-plugin-0.5.1/panel-plugin/libapm.h
--- xfce4-battery-plugin-0.5.1.orig/panel-plugin/libapm.h	2008-09-04 15:53:40.000000000 -0500
+++ xfce4-battery-plugin-0.5.1/panel-plugin/libapm.h	2008-09-09 22:39:46.536365973 -0500
@@ -17,8 +17,8 @@
  * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <linux/apm_bios.h>
 #include <sys/types.h>
+#include <linux/apm_bios.h>
 
 #define APM_PROC   "/proc/apm"
 #define APM_DEVICE "/dev/apm_bios"
