--- llvm-2.6/Makefile.rules.timestamp	2009-08-19 18:04:44.000000000 -0400
+++ llvm-2.6/Makefile.rules	2009-09-09 02:10:38.287389725 -0400
@@ -672,7 +672,7 @@
 
 ProgInstall   = $(INSTALL) $(Install.StripFlag) -m 0755
 ScriptInstall = $(INSTALL) -m 0755
-DataInstall   = $(INSTALL) -m 0644
+DataInstall   = $(INSTALL) -p -m 0644
 
 # When compiling under Mingw/Cygwin, the tblgen tool expects Windows
 # paths. In this case, the SYSPATH function (defined in
