diff -rupN lightdm-1.16.7.orig/data/lightdm.conf lightdm-1.16.7/data/lightdm.conf
--- lightdm-1.16.7.orig/data/lightdm.conf	2016-02-07 10:44:47.229197400 +0100
+++ lightdm-1.16.7/data/lightdm.conf	2016-02-07 10:47:36.011855600 +0100
@@ -21,13 +21,13 @@
 #start-default-seat=true
 #greeter-user=lightdm
 #minimum-display-number=0
-#minimum-vt=7
+#minimum-vt=7 # Setting this to a value < 7 implies security issues, see FS#46799
 #lock-memory=true
 #user-authority-in-system-dir=false
 #guest-account-script=guest-account
 #logind-check-graphical=false
 #log-directory=/var/log/lightdm
-#run-directory=/var/run/lightdm
+run-directory=/run/lightdm
 #cache-directory=/var/cache/lightdm
 #sessions-directory=/usr/share/lightdm/sessions:/usr/share/xsessions:/usr/share/wayland-sessions
 #remote-sessions-directory=/usr/share/lightdm/remote-sessions
@@ -110,7 +110,7 @@
 #allow-user-switching=true
 #allow-guest=true
 #guest-session=
-#session-wrapper=lightdm-session
+session-wrapper=/etc/lightdm/Xsession
 #greeter-wrapper=
 #guest-wrapper=
 #display-setup-script=
diff -rupN lightdm-1.16.7.orig/data/users.conf lightdm-1.16.7/data/users.conf
--- lightdm-1.16.7.orig/data/users.conf	2016-02-07 10:44:47.224423000 +0100
+++ lightdm-1.16.7/data/users.conf	2016-02-07 10:44:53.660216200 +0100
@@ -9,6 +9,6 @@
 # hidden-shells = Shells that indicate a user cannot login
 #
 [UserList]
-minimum-uid=500
+minimum-uid=1000
 hidden-users=nobody nobody4 noaccess
-hidden-shells=/bin/false /usr/sbin/nologin /sbin/nologin
+hidden-shells=/bin/false /usr/bin/nologin /bin/nologin
