diff -up lightdm-1.2.2/src/lightdm.c.nodaemon_option lightdm-1.2.2/src/lightdm.c
--- lightdm-1.2.2/src/lightdm.c.nodaemon_option	2012-04-18 18:29:23.000000000 -0500
+++ lightdm-1.2.2/src/lightdm.c	2012-04-25 14:58:20.947571800 -0500
@@ -883,7 +883,7 @@ main (int argc, char **argv)
         g_printerr (/* Text printed out when an unknown command-line argument provided */
                     _("Run '%s --help' to see a full list of available command line options."), argv[0]);
         g_printerr ("\n");
-        return EXIT_FAILURE;
+    /*  return EXIT_FAILURE; */
     }
 
     if (show_version)
