--- a/src/dhcp-manager/nm-dhcp-client.c	2015-05-27 07:01:42.332134160 +0000
+++ b/src/dhcp-manager/nm-dhcp-client.c	2015-05-27 07:02:36.225520909 +0000
@@ -1542,7 +1542,11 @@
 		g_hash_table_destroy (priv->options);
 		priv->options = NULL;
 	}
-	g_clear_pointer (&priv->iface, g_free);
+
+	if (priv->iface) {
+		g_free(priv->iface);
+		priv->iface = NULL;
+	}
 
 	if (priv->hwaddr) {
 		g_byte_array_free (priv->hwaddr, TRUE);
