--- src/dhcp-manager/nm-dhcp-dhclient-utils.c.orig	2014-03-26 04:47:38.582398880 +0000
+++ src/dhcp-manager/nm-dhcp-dhclient-utils.c	2014-03-26 04:48:41.204603895 +0000
@@ -200,6 +200,7 @@
 	add_also_request (alsoreq, "ms-classless-static-routes");
 	add_also_request (alsoreq, "wpad");
 	add_also_request (alsoreq, "ntp-servers");
+	add_also_request (alsoreq, "tftp-server-name");
 
 	/* And add it to the dhclient configuration */
 	for (i = 0; i < alsoreq->len; i++) {
