--- src/dhcp-manager/nm-dhcp-dhclient-utils.c.orig	2014-06-10 15:04:20.506656249 +0000
+++ src/dhcp-manager/nm-dhcp-dhclient-utils.c	2014-06-10 15:05:04.025339204 +0000
@@ -234,6 +234,7 @@
 		add_also_request (alsoreq, "static-routes");
 		add_also_request (alsoreq, "wpad");
 		add_also_request (alsoreq, "ntp-servers");
+		add_also_request (alsoreq, "tftp-server-name");
 	}
 
 	/* And add it to the dhclient configuration */
