--- src/udev/udevd.c.orig	2017-07-17 15:26:27.797959637 +0000
+++ src/udev/udevd.c	2017-07-17 15:26:53.496083352 +0000
@@ -1731,8 +1731,6 @@
         if (arg_daemonize) {
                 pid_t pid;
 
-                log_info("starting version " VERSION);
-
                 /* connect /dev/null to stdin, stdout, stderr */
                 if (log_get_max_level() < LOG_DEBUG)
                         (void) make_null_stdio();
