diff -rupN xf86-video-ark-0.7.5-orig/src/ark_driver.c xf86-video-ark-0.7.5/src/ark_driver.c
--- xf86-video-ark-0.7.5-orig/src/ark_driver.c	2013-03-09 14:59:45.471029330 +0100
+++ xf86-video-ark-0.7.5/src/ark_driver.c	2013-03-09 15:00:59.248549870 +0100
@@ -39,7 +39,6 @@
 #include "compiler.h"
 #include "mipointer.h"
 #include "micmap.h"
-#include "mibstore.h"
 #include "fb.h"
 #include "ark.h"
 
@@ -538,7 +537,6 @@ static Bool ARKScreenInit(SCREEN_INIT_AR
 
 	fbPictureInit (pScreen, 0, 0);
 
-	miInitializeBackingStore(pScreen);
 	xf86SetBackingStore(pScreen);
 
 	if (!pARK->NoAccel) {
