diff -urN go.orig/src/crypto/x509/root_linux.go go/src/crypto/x509/root_linux.go
--- go.orig/src/crypto/x509/root_linux.go	2017-02-16 13:12:23.000000000 -0600
+++ go/src/crypto/x509/root_linux.go	2017-02-17 15:41:47.471727541 -0600
@@ -11,4 +11,5 @@
 	"/etc/ssl/ca-bundle.pem",                            // OpenSUSE
 	"/etc/pki/tls/cacert.pem",                           // OpenELEC
 	"/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem", // CentOS/RHEL 7
+	"/etc/ssl/cert.pem",                                 // CRUX
 }
