diff -Nru pcmciautils-017-orig/Makefile pcmciautils-017/Makefile
--- pcmciautils-017-orig/Makefile	2010-07-01 23:11:43.000000000 +0300
+++ pcmciautils-017/Makefile	2010-07-01 23:13:52.000000000 +0300
@@ -66,7 +66,7 @@
 exec_prefix =	${prefix}
 etcdir =	${prefix}/etc
 sbindir =	${exec_prefix}/sbin
-mandir =	${prefix}/usr/share/man
+mandir =	${prefix}/usr/man
 udevdir =	${prefix}/lib/udev
 
 INSTALL = /usr/bin/install -c
@@ -79,7 +79,7 @@
 hotplugdir =	${etcdir}/hotplug
 
 # place to put our udev rules to
-udevrulesdir = 	${udevdir}/rules.d
+udevrulesdir = 	${etcdir}/udev/rules.d
 
 # place to put our udev helper binaries (pcmcia-socket-startup, pcmcia-check-broken-cis) to
 udevhelperdir =   ${udevdir}
