From Rawhide. Updates mii ioctls.

diff -Naur net-tools-1.60.orig/include/mii.h net-tools-1.60/include/mii.h
--- net-tools-1.60.orig/include/mii.h	2000-05-21 14:21:27.000000000 +0000
+++ net-tools-1.60/include/mii.h	2003-09-12 22:47:27.000000000 +0000
@@ -11,11 +11,9 @@
 
 /* network interface ioctl's for MII commands */
 #ifndef SIOCGMIIPHY
-#define SIOCGMIIPHY (SIOCDEVPRIVATE)	/* Read from current PHY */
-#define SIOCGMIIREG (SIOCDEVPRIVATE+1) 	/* Read any PHY register */
-#define SIOCSMIIREG (SIOCDEVPRIVATE+2) 	/* Write any PHY register */
-#define SIOCGPARAMS (SIOCDEVPRIVATE+3) 	/* Read operational parameters */
-#define SIOCSPARAMS (SIOCDEVPRIVATE+4) 	/* Set operational parameters */
+#define SIOCGMIIPHY 0x8947 	/* Read from current PHY */
+#define SIOCGMIIREG 0x8948 	/* Read any PHY register */
+#define SIOCSMIIREG 0x8949 	/* Write any PHY register */
 #endif
 
 #include <linux/types.h>
