From Rawhide. Allows check of multiple devices.

diff -Naur net-tools-1.60.orig/lib/nstrcmp.c net-tools-1.60/lib/nstrcmp.c
--- net-tools-1.60.orig/lib/nstrcmp.c	1999-01-09 15:55:20.000000000 +0000
+++ net-tools-1.60/lib/nstrcmp.c	2003-09-12 22:46:13.000000000 +0000
@@ -16,8 +16,8 @@
 	b++;
     }
     if (isdigit(*a)) {
-	if (!isdigit(*b))
-	    return -1;
+        if (!isdigit(*b))
+	    return 1;
 	while (a > astr) {
 	    a--;
 	    if (!isdigit(*a)) {
