--- Makefile.am.orig	2015-03-11 00:49:34.169622639 +0000
+++ Makefile.am	2015-03-11 00:50:08.805743782 +0000
@@ -147,7 +147,7 @@
 .DELETE_ON_ERROR: $(gummboot_solib)
 
 $(gummiboot): $(gummiboot_solib)
-	$(AM_V_GEN) objcopy -j .text -j .sdata -j .data -j .dynamic \
+	$(AM_V_GEN) ${CROSS_COMPILE}objcopy -j .text -j .sdata -j .data -j .dynamic \
 	  -j .dynsym -j .rel -j .rela -j .reloc \
 	  --target=efi-app-$(ARCH) $< $@
 
@@ -182,7 +182,7 @@
 .DELETE_ON_ERROR: $(gummboot_solib)
 
 $(stub): $(stub_solib)
-	$(AM_V_GEN) objcopy -j .text -j .sdata -j .data -j .dynamic \
+	$(AM_V_GEN) ${CROSS_COMPILE}objcopy -j .text -j .sdata -j .data -j .dynamic \
 	  -j .dynsym -j .rel -j .rela -j .reloc \
 	  --target=efi-app-$(ARCH) $< $@
 
