--- a/configure	2014-06-02 20:00:20.582809617 +0000
+++ b/configure	2014-06-02 20:00:58.303276084 +0000
@@ -6161,8 +6161,8 @@
         cat > $TMPC << EOF
 #include <stdio.h>
 #include <cdio/version.h>
-#include <cdio/cdda.h>
-#include <cdio/paranoia.h>
+#include <cdio/paranoia/cdda.h>
+#include <cdio/paranoia/paranoia.h>
 int main(void) {
     void *test = cdda_verbose_set;
     printf("%s\n", CDIO_VERSION);
--- a/stream/stream_cdda.c	2014-06-02 20:01:43.242023291 +0000
+++ b/stream/stream_cdda.c	2014-06-02 20:02:08.512005594 +0000
@@ -21,8 +21,8 @@
 #include <cdda_interface.h>
 #include <cdda_paranoia.h>
 #else
-#include <cdio/cdda.h>
-#include <cdio/paranoia.h>
+#include <cdio/paranoia/cdda.h>
+#include <cdio/paranoia/paranoia.h>
 #endif
 #include <stdio.h>
 #include <stdlib.h>
