--- libbb/find_pid_by_name.c.old	2012-07-27 22:13:13.337183830 +0000
+++ libbb/find_pid_by_name.c	2012-07-27 22:13:53.169685855 +0000
@@ -92,6 +92,7 @@
 		 || (p->argv0 && strcmp(bb_basename(p->argv0), procName) == 0)
 		/* or we require /proc/PID/exe link to match */
 		 || (p->exe && strcmp(bb_basename(p->exe), procName) == 0)
+		 || (p->exe && strcmp(p->exe, procName) == 0)
 		) {
 			pidList = xrealloc_vector(pidList, 2, i);
 			pidList[i++] = p->pid;
