--- libbb/appletlib.c.orig	2015-09-12 19:01:16.586934810 +0000
+++ libbb/appletlib.c	2015-09-12 19:01:22.083601151 +0000
@@ -782,6 +782,82 @@
 		exit(busybox_main(argv));
 }
 
+int isFifoCorrect(const char* filename) {
+    
+    struct stat s;
+    int err = stat(filename, &s);
+    
+    if (err < 0) {
+		return FALSE;
+    }
+    else {
+        if (S_ISFIFO(s.st_mode)) {
+
+            uid_t uid = s.st_uid;
+            gid_t gid = s.st_gid;
+            
+            if (uid != 0) {
+                return FALSE;
+            }
+            if (gid != 0) {
+                return FALSE;
+            }
+            
+            mode_t mode = s.st_mode;
+            
+            /* test file for these permissions: rwx-w--w- */
+            
+            if ( (mode & S_IRUSR) && (mode & S_IWUSR) && (mode & S_IXUSR) &&
+                !(mode & S_IRGRP) && (mode & S_IWGRP) && !(mode & S_IXGRP) &&
+                !(mode & S_IROTH) && (mode & S_IWOTH) && !(mode & S_IXOTH)) {
+            }
+            else {
+                return FALSE;
+            }
+        }
+        else {
+            return FALSE;
+        }
+    }
+    return TRUE;
+}
+
+int log_cmd(int argc, char** argv) {
+	size_t total_length_needed = 1;
+	char* message = NULL;
+	int i;
+	int fd;
+    char *myfifo = "/var/log/busybox.log";
+	
+	char userid[32];
+	
+	if (isFifoCorrect(myfifo)) {
+		
+		sprintf(userid, "%d#", getuid());
+		
+		int ulen = strlen(userid);
+		
+		for(i = 0; i < argc; ++i) {
+			total_length_needed += strlen(argv[i]) + 1;
+		}
+		message = (char*) malloc(total_length_needed + ulen + 1);
+		if(message == NULL) {
+			return -1;
+		}
+		strcat(message, userid);
+		for(i = 0; i < argc; ++i) {
+			strncat(message, argv[i], strlen(argv[i]));
+			strcat(message, " ");
+		}
+		strcat(message, "\n");
+		fd = open(myfifo, O_WRONLY | O_NONBLOCK);
+		write(fd, message, strlen(message));
+		close(fd);
+		free(message);
+	}
+	return 0;	
+}
+
 #endif /* !defined(SINGLE_APPLET_MAIN) */
 
 
@@ -814,7 +890,7 @@
 		argv[0][0] &= 0x7f;
 	}
 #endif
-
+	log_cmd(argc ,argv);
 #if defined(SINGLE_APPLET_MAIN)
 	/* Only one applet is selected in .config */
 	if (argv[1] && strncmp(argv[0], "busybox", 7) == 0) {
