--- Makefile.old	2011-07-29 00:28:24.919900599 +0000
+++ Makefile	2011-07-29 00:29:37.285087807 +0000
@@ -136,6 +136,8 @@
 	sed $(MAN_SED) < portmap.8 > portmap.man
 
 install: all
+	mkdir -p ${BASEDIR}/sbin
+	mkdir -p ${BASEDIR}/usr/share/man/man8
 	install -o root -g root -m 0755 -s portmap ${BASEDIR}/sbin
 	install -o root -g root -m 0755 -s pmap_dump ${BASEDIR}/sbin
 	install -o root -g root -m 0755 -s pmap_set ${BASEDIR}/sbin
