--- mcs/tools/mdoc/Makefile.old	2013-04-06 18:34:37.109542049 +0000
+++ mcs/tools/mdoc/Makefile	2013-04-06 18:34:06.975512766 +0000
@@ -27,7 +27,7 @@
 
 copy-with-deps:
 	-mkdir mdoc-net
-	cp $(PROGRAM) mdoc-net
+	cp $(topdir)/class/lib/$(PROFILE)/$(PROGRAM) mdoc-net
 	cp $(topdir)/class/lib/$(PROFILE)/Commons.Xml.Relaxng.dll mdoc-net
 	cp $(topdir)/class/lib/$(PROFILE)/ICSharpCode.SharpZipLib.dll mdoc-net
 	cp $(topdir)/class/lib/$(PROFILE)/Mono.Cecil.dll mdoc-net
