diff -Nru gcc-4.7.3.orig/fixincludes/Makefile.in gcc-4.7.3/fixincludes/Makefile.in
--- gcc-4.7.3.orig/fixincludes/Makefile.in	2013-04-13 18:33:41.098439867 +0200
+++ gcc-4.7.3/fixincludes/Makefile.in	2013-04-13 18:41:29.354868695 +0200
@@ -173,9 +173,7 @@
 	cd $(srcdir) && $(ACLOCAL) $(ACLOCAL_AMFLAGS)
 
 check : all
-	autogen -T $(srcdir)/check.tpl $(srcdir)/inclhack.def
-	$(SHELL) ./check.sh $(srcdir)/tests/base
-	@rm -f ./check.sh
+	echo "No check here.."
 
 install : all
 	-rm -rf $(DESTDIR)$(itoolsdir)
