--- a/src/capplet/notification-properties.c	2015-01-28 11:22:43.223977161 +0000
+++ b/src/capplet/notification-properties.c	2015-01-28 11:21:16.994582703 +0000
@@ -434,8 +434,7 @@
 
 	dialog->preview = notify_notification_new(_("Notification Test"),
 											  _("Just a test"),
-											  "gnome-util",
-											  NULL);
+											  "gnome-util");
 
 	if (!notify_notification_show(dialog->preview, &error))
 	{
