diff -rupN xf86-video-tga-1.2.2-orig/src/tga_accel.c xf86-video-tga-1.2.2/src/tga_accel.c
--- xf86-video-tga-1.2.2-orig/src/tga_accel.c	2013-03-09 15:25:10.281170192 +0100
+++ xf86-video-tga-1.2.2/src/tga_accel.c	2013-03-09 15:25:33.688970603 +0100
@@ -37,7 +37,6 @@
 #include "xf86cmap.h"
 #include "mipointer.h"
 
-#include "mibstore.h"
 #include "miline.h"
 
 #include "tga_regs.h"
diff -rupN xf86-video-tga-1.2.2-orig/src/tga_driver.c xf86-video-tga-1.2.2/src/tga_driver.c
--- xf86-video-tga-1.2.2-orig/src/tga_driver.c	2013-03-09 15:25:10.281170192 +0100
+++ xf86-video-tga-1.2.2/src/tga_driver.c	2013-03-09 15:25:46.120395683 +0100
@@ -46,8 +46,6 @@
 
 /* software cursor */
 #include "mipointer.h"
-/* backing store */
-#include "mibstore.h"
 
 /*  #include "mibank.h" */
 /* colormap manipulation */
@@ -1451,7 +1449,6 @@ TGAScreenInit(SCREEN_INIT_ARGS_DECL)
     
     fbPictureInit (pScreen, 0, 0);
     
-    miInitializeBackingStore(pScreen);
     xf86SetBackingStore(pScreen);
     xf86SetSilkenMouse(pScreen);
 
