--- a/installer.sh	2013-08-22 22:26:38.129996372 +0000
+++ b/installer.sh	2013-08-22 22:31:33.126285860 +0000
@@ -6,14 +6,14 @@
 #
 
 # Target directories (can be changed to customize the installation)
-INSTALLDIR=/usr/local/usb-redirector  # root dir where the software will be installed
+INSTALLDIR=$PKG  # root dir where the software will be installed
 FILESDIR=files                     # directory name where installer files are located
 MODULESBINDIR=modules/bin          # directory name where compiled stub driver modules are located
 MODULESSRCDIR=modules/src          # directory name where stub driver sources are located
-SHELLLINKDIR=/usr/local/bin        # directory to place symbolic link for shell executable
+SHELLLINKDIR=$PKG/usr/local/bin        # directory to place symbolic link for shell executable
 
 # Default kernel sources path
 KERNELDIR=/lib/modules/`uname -r`/build
 
 # How to install kernel sources (or kernel headers).
 # Debian,Ubuntu: apt-get install make gcc linux-headers-`uname -r`
@@ -242,7 +242,7 @@
     SYS_DISTRIB_MODVER=no
     if [ -d /etc/init.d ]; then
       SYS_DISTRIB_INIT_TYPE=sysv
-      SYS_DISTRIB_INIT_DIR=/etc/init.d
+      SYS_DISTRIB_INIT_DIR=$PKG/etc/init.d
       SYS_DISTRIB_RUNLEVEL_DIR=/etc
     elif [ -d /etc/rc.d/init.d ]; then
       SYS_DISTRIB_INIT_TYPE=sysv
