diff --git a/src/installtransaction.cpp b/src/installtransaction.cpp
index 849931a..70e5e37 100644
--- a/src/installtransaction.cpp
+++ b/src/installtransaction.cpp
@@ -658,5 +658,7 @@ string InstallTransaction::getPkgmkPackageDir()
 
 string InstallTransaction::getPkgmkCompressionMode()
 {
-    return getPkgmkSetting("PKGMK_COMPRESSION_MODE");
+    string value = getPkgmkSetting("PKGMK_COMPRESSION_MODE");
+
+    return value.size() ? value : "gz";
 }
