--- a/configure	2014-05-09 04:29:48.000000000 +0000
+++ b/configure	2014-06-05 15:33:49.921354596 +0000
@@ -17601,31 +17601,6 @@
 	# Put the nasty error message in config.log where it belongs
 	echo "$WAYLAND_SCANNER_PKG_ERRORS" >&5
 
-	as_fn_error $? "Package requirements (wayland-scanner) were not met:
-
-$WAYLAND_SCANNER_PKG_ERRORS
-
-Consider adjusting the PKG_CONFIG_PATH environment variable if you
-installed software in a non-standard prefix.
-
-Alternatively, you may set the environment variables WAYLAND_SCANNER_CFLAGS
-and WAYLAND_SCANNER_LIBS to avoid the need to call pkg-config.
-See the pkg-config man page for more details." "$LINENO" 5
-elif test $pkg_failed = untried; then
-     	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-	{ { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "The pkg-config script could not be found or is too old.  Make sure it
-is in your PATH or set the PKG_CONFIG environment variable to the full
-path to pkg-config.
-
-Alternatively, you may set the environment variables WAYLAND_SCANNER_CFLAGS
-and WAYLAND_SCANNER_LIBS to avoid the need to call pkg-config.
-See the pkg-config man page for more details.
-
-To get pkg-config, see <http://pkg-config.freedesktop.org/>.
-See \`config.log' for more details" "$LINENO" 5; }
 else
 	WAYLAND_SCANNER_CFLAGS=$pkg_cv_WAYLAND_SCANNER_CFLAGS
 	WAYLAND_SCANNER_LIBS=$pkg_cv_WAYLAND_SCANNER_LIBS
