--- a/Makefile.PL	2013-12-06 17:23:10.425961004 +0000
+++ b/Makefile.PL	2013-12-06 17:23:22.121416273 +0000
@@ -31,7 +31,7 @@
       $script .= <<"INSTALL";
 
 install_sax_expat :
-\t\@\$(PERL) -I\$(INSTALLSITELIB) -MXML::SAX -e "XML::SAX->add_parser(q(XML::SAX::Expat))->save_parsers()"
+\t\@\$(PERL) -I\$(DESTINSTALLSITELIB) -MXML::SAX -e "XML::SAX->add_parser(q(XML::SAX::Expat))->save_parsers()"
 
 INSTALL
     } else {
