diff -aur btrfs-progs-unstable-075587c.orig/btrfsck.c btrfs-progs-unstable-075587c/btrfsck.c
--- btrfs-progs-unstable-075587c.orig/btrfsck.c	2010-04-06 15:30:17.000000000 +0200
+++ btrfs-progs-unstable-075587c/btrfsck.c	2010-08-11 08:32:57.623000001 +0200
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <fcntl.h>
+#include <sys/stat.h>
 #include "kerncompat.h"
 #include "ctree.h"
 #include "disk-io.h"
