--- lib/hcrypto/Makefile.in.orig	2020-10-31 21:07:10.292281787 +0000
+++ lib/hcrypto/Makefile.in	2020-10-31 21:08:02.167716272 +0000
@@ -3553,7 +3553,7 @@
 	  $(check_SCRIPTS)
 	$(MAKE) $(AM_MAKEFLAGS) check-TESTS check-local
 check: check-am
-all-am: Makefile $(PROGRAMS) $(LTLIBRARIES) $(HEADERS) all-local
+all-am: Makefile hcrypto-link $(PROGRAMS) $(LTLIBRARIES) $(HEADERS) all-local
 installdirs:
 	for dir in "$(DESTDIR)$(libdir)" "$(DESTDIR)$(hcryptoincludedir)"; do \
 	  test -z "$$dir" || $(MKDIR_P) "$$dir"; \
