diff -Nru gcc-4.2.1.orig/fixincludes/Makefile.in gcc-4.2.1/fixincludes/Makefile.in
--- gcc-4.2.1.orig/fixincludes/Makefile.in	2007-08-15 13:45:59.000000000 +0200
+++ gcc-4.2.1/fixincludes/Makefile.in	2007-08-15 14:21:16.000000000 +0200
@@ -169,9 +169,7 @@
 	cp $(srcdir)/../gcc/aclocal.m4 .
 
 check : all
-	autogen -T $(srcdir)/check.tpl $(srcdir)/inclhack.def
-	$(SHELL) ./check.sh $(srcdir)/tests/base
-	@rm -f ./check.sh
+	echo "No check here.."
 
 install : all
 	-rm -rf $(DESTDIR)$(itoolsdir)
