diff -ur idesk.old/src/DesktopConfig.cpp idesk-0.7.5/src/DesktopConfig.cpp
--- idesk.old/src/DesktopConfig.cpp	2011-05-18 21:04:57.000000000 +0000
+++ idesk-0.7.5/src/DesktopConfig.cpp	2011-05-18 21:35:05.000000000 +0000
@@ -24,6 +24,7 @@
 
 #include "DesktopConfig.h"
 #include "Util.h"
+#include <sys/stat.h>
 
 //the initilizer list just sets the program defaults for non-necessary options
 DesktopConfig::DesktopConfig(Database db, string ideskrcFile) :
