diff -rupN xf86-video-fbdev-0.4.3-orig/src/fbdev.c xf86-video-fbdev-0.4.3/src/fbdev.c
--- xf86-video-fbdev-0.4.3-orig/src/fbdev.c	2013-03-09 15:15:59.833345651 +0100
+++ xf86-video-fbdev-0.4.3/src/fbdev.c	2013-03-09 15:16:21.810097323 +0100
@@ -14,7 +14,6 @@
 #include "xf86_OSproc.h"
 
 #include "mipointer.h"
-#include "mibstore.h"
 #include "micmap.h"
 #include "colormapst.h"
 #include "xf86cmap.h"
@@ -845,7 +844,6 @@ FBDevScreenInit(SCREEN_INIT_ARGS_DECL)
 	}
 
 	xf86SetBlackWhitePixels(pScreen);
-	miInitializeBackingStore(pScreen);
 	xf86SetBackingStore(pScreen);
 
 	/* software cursor */
