--- src/notification/notification-main.c.orig	2014-04-18 18:46:51.434238028 +0000
+++ src/notification/notification-main.c	2014-04-18 18:48:36.685375134 +0000
@@ -535,8 +535,7 @@
                     _("A hard disk may be failing"),
                     /* Translators: This is used as the text of the notification*/
                     _("One or more hard disks report health problems. Click the icon to get more information."),
-                    "gtk-dialog-warning",
-                    NULL);
+                    "gtk-dialog-warning");
                 notify_notification_attach_to_status_icon (data->ata_smart_notification,
                                                            data->status_icon);
                 notify_notification_set_urgency (data->ata_smart_notification, NOTIFY_URGENCY_CRITICAL);
--- src/notification/notification-main.c.orig	2014-04-18 19:13:16.236999415 +0000
+++ src/notification/notification-main.c	2014-04-18 19:13:27.858451606 +0000
@@ -536,8 +536,6 @@
                     /* Translators: This is used as the text of the notification*/
                     _("One or more hard disks report health problems. Click the icon to get more information."),
                     "gtk-dialog-warning");
-                notify_notification_attach_to_status_icon (data->ata_smart_notification,
-                                                           data->status_icon);
                 notify_notification_set_urgency (data->ata_smart_notification, NOTIFY_URGENCY_CRITICAL);
                 notify_notification_set_timeout (data->ata_smart_notification, NOTIFY_EXPIRES_NEVER);
                 show_notification (data);
