--- a/src/ttf.h	2011-06-12 21:43:39.000000000 +0000
+++ b/src/ttf.h	2014-01-05 21:50:28.986182005 +0000
@@ -1,8 +1,8 @@
 #ifndef _TTF_H
 #define _TTF_H
 #include <ft2build.h>
-#include <freetype/ftoutln.h>
-#include <freetype/ttnameid.h>
+#include FT_OUTLINE_H
+#include FT_TRUETYPE_IDS_H
 
 #define CACHED_METRICS  0x10
 #define CACHED_BITMAP   0x01
--- a/src/ttf.c	2011-06-12 21:43:39.000000000 +0000
+++ b/src/ttf.c	2014-01-05 21:51:18.924060559 +0000
@@ -25,8 +25,8 @@
 #include <assert.h>
 
 #include <ft2build.h>
-#include <freetype/ftoutln.h>
-#include <freetype/ttnameid.h>
+#include FT_OUTLINE_H
+#include FT_TRUETYPE_IDS_H
 
 #include "common.h"
 #include "render.h"
