diff -Nru tar-1.18.orig/src/extract.c tar-1.18/src/extract.c
--- tar-1.18.orig/src/extract.c	2007-07-05 08:19:23.000000000 +0200
+++ tar-1.18/src/extract.c	2007-07-05 08:19:37.000000000 +0200
@@ -113,8 +113,6 @@
 extr_init (void)
 {
   we_are_root = geteuid () == 0;
-  same_permissions_option += we_are_root;
-  same_owner_option += we_are_root;
 
   /* Option -p clears the kernel umask, so it does not affect proper
      restoration of file permissions.  New intermediate directories will
