diff -Nru samba-3.0.26a.orig/examples/smb.conf.default samba-3.0.26a/examples/smb.conf.default
--- samba-3.0.26a.orig/examples/smb.conf.default	2007-09-12 09:39:16.000000000 +0200
+++ samba-3.0.26a/examples/smb.conf.default	2007-09-12 09:39:40.000000000 +0200
@@ -63,7 +63,7 @@
 
 # this tells Samba to use a separate log file for each machine
 # that connects
-   log file = /usr/local/samba/var/log.%m
+   log file = /var/log/samba/log.%m
 
 # Put a capping on the size of the log files (in Kb).
    max log size = 50
@@ -184,7 +184,7 @@
 # specifically define each individual printer
 [printers]
    comment = All Printers
-   path = /usr/spool/samba
+   path = /var/spool/samba
    browseable = no
 # Set public = yes to allow user 'guest account' to print
    guest ok = no
