--- a/src/nma-pkcs11-token-login-dialog.h	2020-05-13 17:33:50.491411868 +0000
+++ b/src/nma-pkcs11-token-login-dialog.h	2020-05-13 17:34:10.291086416 +0000
@@ -20,7 +20,7 @@
 } NMAPkcs11TokenLoginDialog;
 
 typedef struct {
-        GtkDialog parent;
+        GtkDialogClass parent;
 } NMAPkcs11TokenLoginDialogClass;
 
 #define NMA_TYPE_PKCS11_TOKEN_LOGIN_DIALOG            (nma_pkcs11_token_login_dialog_get_type ())
