diff -Nru cups-1.2.0.orig/conf/cupsd.conf.in cups-1.2.0/conf/cupsd.conf.in
--- cups-1.2.0.orig/conf/cupsd.conf.in	2006-05-14 10:35:54.000000000 +0200
+++ cups-1.2.0/conf/cupsd.conf.in	2006-05-14 10:36:54.000000000 +0200
@@ -10,6 +10,10 @@
 # troubleshooting...
 LogLevel info
 
+# Location of ssl certificates
+ServerCertificate /etc/ssl/certs/cups.crt
+ServerKey /etc/ssl/keys/cups.key
+
 # Administrator user group...
 SystemGroup @CUPS_SYSTEM_GROUPS@
 
