--- chrome/browser/ui/startup/bad_flags_prompt.cc.orig	2017-07-23 23:11:49.972153525 +0000
+++ chrome/browser/ui/startup/bad_flags_prompt.cc	2017-07-23 23:13:16.420794199 +0000
@@ -44,15 +44,6 @@
   // security will suffer".
   static const char* kBadFlags[] = {
     // These flags disable sandbox-related security.
-    switches::kDisableGpuSandbox,
-    switches::kDisableSeccompFilterSandbox,
-    switches::kDisableSetuidSandbox,
-    switches::kDisableWebSecurity,
-#if !defined(DISABLE_NACL)
-    switches::kNaClDangerousNoSandboxNonSfi,
-#endif
-    switches::kNoSandbox,
-    switches::kSingleProcess,
 
     // These flags disable or undermine the Same Origin Policy.
     translate::switches::kTranslateSecurityOrigin,
