--- config.status.orig	2014-01-27 17:11:35.470140655 +0000
+++ config.status	2014-01-27 17:12:58.125084307 +0000
@@ -639,20 +639,12 @@
 S["am__EXEEXT_TRUE"]="#"
 S["LTLIBOBJS"]=""
 S["LIBOBJS"]=""
-S["VTE_LIBS"]="-lvte -lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lgdk_pixbuf-2.0 -lpangoft2-1.0 -lpango-1.0 -lfreetype -lfontconfig -lgio-2.0 -lgobject"\
-"-2.0 -lglib-2.0 -lcairo -lX11 -lXext "
-S["VTE_CFLAGS"]="-pthread -I/usr/include/vte-0.0 -I/usr/include/gtk-2.0 -I/usr/lib/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/gdk"\
-"-pixbuf-2.0 -I/usr/include/libpng14 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/gio-unix-2.0/ -I/usr/in"\
-"clude/cairo -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libdrm -I/usr/includ"\
-"e/libpng14 "
-S["HAVE_VTE"]="1"
-S["GLADE_LIB_LIBS"]="-lglade-2.0 -lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lgli"\
-"b-2.0 -lfreetype -lfontconfig -lxml2 "
-S["GLADE_LIB_CFLAGS"]="-pthread -I/usr/include/libglade-2.0 -I/usr/include/gtk-2.0 -I/usr/lib/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/includ"\
-"e/cairo -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/libpng14 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng14 -I/usr/include/p"\
-"ango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/freetype2 -I/usr/includ"\
-"e/libxml2 "
-S["HAVE_GLADE_LIB"]="1"
+S["VTE_LIBS"]=""
+S["VTE_CFLAGS"]=""
+S["HAVE_VTE"]="0"
+S["GLADE_LIB_LIBS"]=""
+S["GLADE_LIB_CFLAGS"]=""
+S["HAVE_GLADE_LIB"]="0"
 S["GTHREAD_LIBS"]="-lgthread-2.0 -pthread -lglib-2.0 "
 S["GTHREAD_CFLAGS"]="-pthread -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include "
 S["GTK_LIBS"]="-lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lfree"\
