diff -ur a/configure b/configure
--- a/configure	2013-01-06 15:01:36.000000000 +0000
+++ b/configure	2014-02-07 17:00:07.340938320 +0000
@@ -5366,7 +5366,7 @@
 
 glade_lib="libglade-2.0"
 
-pkg_failed=no
+pkg_failed=yes
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for GLADE_LIB" >&5
 $as_echo_n "checking for GLADE_LIB... " >&6; }
 
@@ -5453,7 +5453,7 @@
 
 vte_lib="vte >= 0.23.5"
 
-pkg_failed=no
+pkg_failed=yes
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for VTE" >&5
 $as_echo_n "checking for VTE... " >&6; }
 
