--- config.status.orig	2014-01-25 16:44:29.933655823 +0000
+++ config.status	2014-01-25 16:45:43.906177549 +0000
@@ -639,18 +639,12 @@
 S["am__EXEEXT_TRUE"]="#"
 S["LTLIBOBJS"]=""
 S["LIBOBJS"]=""
-S["VTE_LIBS"]="-lvte -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lpangoft2-1.0 -lpangocairo-1.0 -lgdk_pixbuf-2.0 -lpango-1.0 -lfreetype -lfontconfig -lgio-2.0 -lgobject"\
-"-2.0 -lglib-2.0 -lcairo -lX11 -lXext  "
-S["VTE_CFLAGS"]="-pthread -I/usr/include/vte-0.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/pango-1.0 -I/usr/include/gtk-2.0 -I/usr/include/g"\
-"io-unix-2.0/ -I/usr/include/cairo -I/usr/lib/gtk-2.0/include -I/usr/include/atk-1.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/pixman-1 -I/usr/inc"\
-"lude/freetype2 -I/usr/include/libpng14  "
-S["HAVE_VTE"]="1"
-S["GLADE_LIB_LIBS"]="-lglade-2.0 -lgtk-x11-2.0 -lxml2 -lgdk-x11-2.0 -latk-1.0 -lgio-2.0 -lpangoft2-1.0 -lpangocairo-1.0 -lgdk_pixbuf-2.0 -lcairo -lpango-1.0 -lfreetype -"\
-"lfontconfig -lgobject-2.0 -lglib-2.0  "
-S["GLADE_LIB_CFLAGS"]="-pthread -I/usr/include/libglade-2.0 -I/usr/include/gtk-2.0 -I/usr/include/libxml2 -I/usr/lib/gtk-2.0/include -I/usr/include/atk-1.0 -I/usr/include/"\
-"cairo -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/pixman-1 -I/usr/incl"\
-"ude/freetype2 -I/usr/include/libpng14  "
-S["HAVE_GLADE_LIB"]="1"
+S["VTE_LIBS"]=""
+S["VTE_CFLAGS"]=""
+S["HAVE_VTE"]="0"
+S["GLADE_LIB_LIBS"]=""
+S["GLADE_LIB_CFLAGS"]=""
+S["HAVE_GLADE_LIB"]="0"
 S["GTHREAD_LIBS"]="-pthread -lgthread-2.0 -lrt -lglib-2.0  "
 S["GTHREAD_CFLAGS"]="-pthread -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include  "
 S["GTK_LIBS"]="-lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgio-2.0 -lpangoft2-1.0 -lpangocairo-1.0 -lgdk_pixbuf-2.0 -lcairo -lpango-1.0 -lfreetype -lfontconfig -lgobje"\
