--- config.status.orig	2014-05-19 19:35:39.730471658 +0000
+++ config.status	2014-05-19 19:36:44.703772602 +0000
@@ -639,12 +639,9 @@
 S["am__EXEEXT_TRUE"]="#"
 S["LTLIBOBJS"]=""
 S["LIBOBJS"]=""
-S["VTE_LIBS"]="-lvte -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lpangoft2-1.0 -lpangocairo-1.0 -lgdk_pixbuf-2.0 -lpango-1.0 -lfreetype -lfontconfig -lgio-2.0 -lgobject"\
-"-2.0 -lglib-2.0 -lcairo -lX11 -lXext  "
-S["VTE_CFLAGS"]="-pthread -I/usr/include/vte-0.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/pango-1.0 -I/usr/include/gtk-2.0 -I/usr/include/g"\
-"io-unix-2.0/ -I/usr/include/cairo -I/usr/lib/gtk-2.0/include -I/usr/include/atk-1.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/pixman-1 -I/usr/inc"\
-"lude/freetype2 -I/usr/include/libpng14  "
-S["HAVE_VTE"]="1"
+S["VTE_LIBS"]=""
+S["VTE_CFLAGS"]=""
+S["HAVE_VTE"]="0"
 S["GLADE_LIB_LIBS"]="-lglade-2.0 -lgtk-x11-2.0 -lxml2 -lgdk-x11-2.0 -latk-1.0 -lgio-2.0 -lpangoft2-1.0 -lpangocairo-1.0 -lgdk_pixbuf-2.0 -lcairo -lpango-1.0 -lfreetype -"\
 "lfontconfig -lgobject-2.0 -lglib-2.0  "
 S["GLADE_LIB_CFLAGS"]="-pthread -I/usr/include/libglade-2.0 -I/usr/include/gtk-2.0 -I/usr/include/libxml2 -I/usr/lib/gtk-2.0/include -I/usr/include/atk-1.0 -I/usr/include/"\
